{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Doublet simulation and removal using DoubletDecon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://satijalab.org/seurat/articles/get_started.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### import libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Attaching SeuratObject\n",
      "\n",
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style> .container { width:95% !important; } </style>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(Seurat)\n",
    "library(dplyr)\n",
    "library(magrittr)\n",
    "IRdisplay::display_html(\"<style> .container { width:95% !important; } </style>\")\n",
    "library(\"xlsx\")\n",
    "#library(mgsa)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Check data quality"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Saline (control) group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_sal.data <-Read10X(data.dir = \"/LS-SAL-1/raw_feature_bc_matrix\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_sal.data <-Read10X(data.dir = \"/LS-SAL-1/raw_feature_bc_matrix\")\n",
    "colnames(LS_sal.data) = paste0(colnames(LS_sal.data),\"LS_sal\")\n",
    "LS_sal<- CreateSeuratObject(counts = LS_sal.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_sal@meta.data$stim <- \"LS_sal\"\n",
    "\n",
    "mito.features <- grep(pattern = \"^mt-\", x = rownames(x =LS_sal), value = TRUE)\n",
    "percent.mito <- Matrix::colSums(x = GetAssayData(object = LS_sal, slot = 'counts')[mito.features, ]) / Matrix::colSums(x = GetAssayData(object = LS_sal, slot = 'counts'))\n",
    "LS_sal[['percent.mito']] <- percent.mito"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "FeatureScatter(object = LS_sal, feature1 = \"nCount_RNA\", feature2 = \"percent.mito\",pt.size=0.1) \n",
    "hist(LS_sal@meta.data$percent.mito,breaks=seq(0,0.7,0.01))\n",
    "hist(LS_sal@meta.data$nCount_RNA,breaks=seq(0,1000000,300),xlim = c(0,8000))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Chronic morphine group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_mor.data <-Read10X(data.dir = \"/LS-MOR-1/raw_feature_bc_matrix\")\n",
    "colnames(LS_mor.data) = paste0(colnames(LS_mor.data),\"LS_mor\")\n",
    "LS_mor<- CreateSeuratObject(counts = LS_mor.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_mor@meta.data$stim <- \"LS_mor\"\n",
    "\n",
    "mito.features <- grep(pattern = \"^mt-\", x = rownames(x =LS_mor), value = TRUE)\n",
    "percent.mito <- Matrix::colSums(x = GetAssayData(object = LS_mor, slot = 'counts')[mito.features, ]) / Matrix::colSums(x = GetAssayData(object =LS_mor, slot = 'counts'))\n",
    "LS_mor[['percent.mito']] <- percent.mito"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "FeatureScatter(object = LS_mor , feature1 = \"nCount_RNA\", feature2 = \"percent.mito\",pt.size=0.1)\n",
    "hist(LS_mor@meta.data$percent.mito,breaks=seq(0,0.7,0.01))\n",
    "hist(LS_mor@meta.data$nCount_RNA,breaks=seq(0,1000000,300),xlim = c(0,8000))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Precipitated withdrawal group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_nal.data <-Read10X(data.dir = \"/LS-NAL-1/raw_feature_bc_matrix\")\n",
    "colnames(LS_nal.data) = paste0(colnames(LS_nal.data),\"LS_nal\")\n",
    "LS_nal<- CreateSeuratObject(counts = LS_nal.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_nal@meta.data$stim <- \"LS_nal\"\n",
    "\n",
    "mito.features <- grep(pattern = \"^mt-\", x = rownames(x =LS_nal), value = TRUE)\n",
    "percent.mito <- Matrix::colSums(x = GetAssayData(object = LS_nal, slot = 'counts')[mito.features, ]) / Matrix::colSums(x = GetAssayData(object = LS_nal, slot = 'counts'))\n",
    "LS_nal[['percent.mito']] <- percent.mito"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "FeatureScatter(object = LS_nal , feature1 = \"nCount_RNA\", feature2 = \"percent.mito\",pt.size=0.1)\n",
    "hist(LS_nal@meta.data$percent.mito,breaks=seq(0,0.7,0.01))\n",
    "hist(LS_nal@meta.data$nCount_RNA,breaks=seq(0,1000000,300),xlim = c(0,8000))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Acute morphine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_one_mor.data <-Read10X(data.dir = \"/Mor_single/analysis\")\n",
    "colnames(LS_one_mor.data) = paste0(colnames(LS_one_mor.data),\"LS_one_mor\")\n",
    "LS_one_mor<- CreateSeuratObject(counts = LS_one_mor.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_one_mor@meta.data$stim <- \"LS_one_mor\"\n",
    "\n",
    "mito.features <- grep(pattern = \"^mt-\", x = rownames(x =LS_one_mor), value = TRUE)\n",
    "percent.mito <- Matrix::colSums(x = GetAssayData(object = LS_one_mor, slot = 'counts')[mito.features, ]) / Matrix::colSums(x = GetAssayData(object = LS_one_mor, slot = 'counts'))\n",
    "LS_one_mor[['percent.mito']] <- percent.mito"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "head(LS_one_mor@meta.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "FeatureScatter(object = LS_one_mor , feature1 = \"nCount_RNA\", feature2 = \"percent.mito\",pt.size=0.1)\n",
    "hist(LS_one_mor@meta.data$percent.mito,breaks=seq(0,0.7,0.01))\n",
    "hist(LS_one_mor@meta.data$nCount_RNA,breaks=seq(0,1000000,300),xlim = c(0,8000))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Acute naloxone (drug-naive)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_nal_no_mor.data <-Read10X(data.dir = \"/Nal_no_mor/analysis\")\n",
    "colnames(LS_nal_no_mor.data) = paste0(colnames(LS_nal_no_mor.data),\"LS_nal_no_mor\")\n",
    "LS_nal_no_mor<- CreateSeuratObject(counts = LS_nal_no_mor.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_nal_no_mor@meta.data$stim <- \"LS_nal_no_mor\"\n",
    "\n",
    "mito.features <- grep(pattern = \"^mt-\", x = rownames(x =LS_nal_no_mor), value = TRUE)\n",
    "percent.mito <- Matrix::colSums(x = GetAssayData(object = LS_nal_no_mor, slot = 'counts')[mito.features, ]) / Matrix::colSums(x = GetAssayData(object = LS_nal_no_mor, slot = 'counts'))\n",
    "LS_nal_no_mor[['percent.mito']] <- percent.mito"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_nal_no_mor"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Sample normalization, scaling, and dimensionality reduction for each sample separately"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample<-c(\"LS_sal\",\"LS_mor\",\"LS_nal\")\n",
    "sample[i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_sal <- subset(x = LS_sal, subset = nCount_RNA > 500 & nCount_RNA < 25000 & percent.mito < 0.1) # you may change percent.mito from 0.2 to 0.3 \n",
    "\n",
    "LS_sal<- NormalizeData(object = LS_sal,verbose = FALSE) \n",
    "\n",
    "LS_sal<- FindVariableFeatures(object =LS_sal,selection.method = \"vst\", nfeatures = 2000, verbose = FALSE)\n",
    "\n",
    "LS_sal<- ScaleData(object = LS_sal, features = rownames(x =LS_sal), vars.to.regress = c(\"nCount_RNA\", \"percent.mito\"))\n",
    "\n",
    "LS_sal<- RunPCA(object = LS_sal, features = VariableFeatures(object =LS_sal), verbose = FALSE)\n",
    "\n",
    "LS_sal<- FindNeighbors(object =LS_sal, dims = 1:30)\n",
    "LS_sal<- FindClusters(object = LS_sal, resolution = 0.8)\n",
    "\n",
    "saveRDS(LS_sal,file = \"/Doublet_processing/LS_sal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_mor <- subset(x = LS_mor, subset = nCount_RNA > 500 & nCount_RNA < 25000 & percent.mito < 0.10) # you may change percent.mito from 0.2 to 0.3 \n",
    "\n",
    "LS_mor<- NormalizeData(object = LS_mor,verbose = FALSE) \n",
    "\n",
    "LS_mor<- FindVariableFeatures(object =LS_mor,selection.method = \"vst\", nfeatures = 2000, verbose = FALSE)\n",
    "\n",
    "LS_mor<- ScaleData(object = LS_mor, features = rownames(x =LS_mor), vars.to.regress = c(\"nCount_RNA\", \"percent.mito\"))\n",
    "\n",
    "LS_mor<- RunPCA(object = LS_mor, features = VariableFeatures(object =LS_mor), verbose = FALSE)\n",
    "\n",
    "LS_mor<- FindNeighbors(object =LS_mor, dims = 1:30)\n",
    "LS_mor<- FindClusters(object = LS_mor, resolution = 0.8)\n",
    "\n",
    "saveRDS(LS_mor,file = \"/Doublet_processing/LS_mor\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_nal <- subset(x = LS_nal, subset = nCount_RNA > 500 & nCount_RNA < 25000 & percent.mito < 0.10) # you may change percent.mito from 0.2 to 0.3 \n",
    "\n",
    "LS_nal<- NormalizeData(object = LS_nal,verbose = FALSE) \n",
    "\n",
    "LS_nal<- FindVariableFeatures(object =LS_nal,selection.method = \"vst\", nfeatures = 2000, verbose = FALSE)\n",
    "\n",
    "LS_nal<- ScaleData(object = LS_nal, features = rownames(x =LS_nal), vars.to.regress = c(\"nCount_RNA\", \"percent.mito\"))\n",
    "\n",
    "LS_nal<- RunPCA(object = LS_nal, features = VariableFeatures(object =LS_nal), verbose = FALSE)\n",
    "\n",
    "LS_nal<- FindNeighbors(object =LS_nal, dims = 1:30)\n",
    "LS_nal<- FindClusters(object = LS_nal, resolution = 0.8)\n",
    "\n",
    "saveRDS(LS_nal,file = \"/Doublet_processing/LS_nal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_one_mor <- subset(x = LS_one_mor, subset = nCount_RNA > 500 & nCount_RNA < 25000 & percent.mito < 0.10) # you may change percent.mito from 0.2 to 0.3 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Regressing out nCount_RNA, percent.mito\n",
      "Centering and scaling data matrix\n",
      "Computing nearest neighbor graph\n",
      "Computing SNN\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck\n",
      "\n",
      "Number of nodes: 8632\n",
      "Number of edges: 301626\n",
      "\n",
      "Running Louvain algorithm...\n",
      "Maximum modularity in 10 random starts: 0.9014\n",
      "Number of communities: 22\n",
      "Elapsed time: 0 seconds\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in gzfile(file, mode):\n",
      "\"cannot open file '/Users/stuberadmin/Documents/scRNAseq/211119-RS/Doublet_processing/LS_one_mor': it is a directory\""
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in gzfile(file, mode): cannot open the connection\n",
     "output_type": "error",
     "traceback": [
      "Error in gzfile(file, mode): cannot open the connection\nTraceback:\n",
      "1. saveRDS(LS_one_mor, file = \"/Users/stuberadmin/Documents/scRNAseq/211119-RS/Doublet_processing/LS_one_mor\")",
      "2. gzfile(file, mode)"
     ]
    }
   ],
   "source": [
    "LS_one_mor<- NormalizeData(object = LS_one_mor,verbose = FALSE) \n",
    "\n",
    "LS_one_mor<- FindVariableFeatures(object =LS_one_mor,selection.method = \"vst\", nfeatures = 2000, verbose = FALSE)\n",
    "\n",
    "LS_one_mor<- ScaleData(object = LS_one_mor, features = rownames(x =LS_one_mor), vars.to.regress = c(\"nCount_RNA\", \"percent.mito\"))\n",
    "\n",
    "LS_one_mor<- RunPCA(object = LS_one_mor, features = VariableFeatures(object =LS_one_mor), verbose = FALSE)\n",
    "\n",
    "LS_one_mor<- FindNeighbors(object =LS_one_mor, dims = 1:30)\n",
    "LS_one_mor<- FindClusters(object = LS_one_mor, resolution = 0.8)\n",
    "\n",
    "saveRDS(LS_one_mor,file = \"/Doublet_processing/LS_one_mor\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(LS_one_mor,file = \"/Doublet_processing/LS_one_mor.RDS\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_nal_no_mor <- subset(x = LS_nal_no_mor, subset = nCount_RNA > 500 & nCount_RNA < 25000 & percent.mito < 0.10) # you may change percent.mito from 0.2 to 0.3 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Regressing out nCount_RNA, percent.mito\n",
      "Centering and scaling data matrix\n",
      "Computing nearest neighbor graph\n",
      "Computing SNN\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck\n",
      "\n",
      "Number of nodes: 9539\n",
      "Number of edges: 371751\n",
      "\n",
      "Running Louvain algorithm...\n",
      "Maximum modularity in 10 random starts: 0.9091\n",
      "Number of communities: 24\n",
      "Elapsed time: 0 seconds\n"
     ]
    }
   ],
   "source": [
    "LS_nal_no_mor <- subset(x = LS_nal_no_mor, subset = nCount_RNA > 500 & nCount_RNA < 25000 & percent.mito < 0.10) # you may change percent.mito from 0.2 to 0.3 \n",
    "\n",
    "LS_nal_no_mor<- NormalizeData(object = LS_nal_no_mor,verbose = FALSE) \n",
    "\n",
    "LS_nal_no_mor<- FindVariableFeatures(object =LS_nal_no_mor,selection.method = \"vst\", nfeatures = 2000, verbose = FALSE)\n",
    "\n",
    "LS_nal_no_mor<- ScaleData(object = LS_nal_no_mor, features = rownames(x =LS_nal_no_mor), vars.to.regress = c(\"nCount_RNA\", \"percent.mito\"))\n",
    "\n",
    "LS_nal_no_mor<- RunPCA(object = LS_nal_no_mor, features = VariableFeatures(object =LS_nal_no_mor), verbose = FALSE)\n",
    "\n",
    "LS_nal_no_mor<- FindNeighbors(object =LS_nal_no_mor, dims = 1:30)\n",
    "LS_nal_no_mor<- FindClusters(object = LS_nal_no_mor, resolution = 0.8)\n",
    "\n",
    "saveRDS(LS_nal_no_mor,file = \"/Doublet_processing/LS_nal_no_mor.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_sal <- RunUMAP(LS_sal, reduction='pca', dims=1:30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_mor <- RunUMAP(LS_mor, reduction='pca', dims=1:30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_nal <- RunUMAP(LS_nal, reduction='pca', dims=1:30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric\n",
      "To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'\n",
      "This message will be shown once per session\"18:28:27 UMAP embedding parameters a = 0.9922 b = 1.112\n",
      "18:28:27 Read 8632 rows and found 30 numeric columns\n",
      "18:28:27 Using Annoy for neighbor search, n_neighbors = 30\n",
      "18:28:27 Building Annoy index with metric = cosine, n_trees = 50\n",
      "0%   10   20   30   40   50   60   70   80   90   100%\n",
      "[----|----|----|----|----|----|----|----|----|----|\n",
      "**************************************************|\n",
      "18:28:29 Writing NN index file to temp file C:\\Users\\STUBER~1\\AppData\\Local\\Temp\\RtmpoF8MPH\\file3b0030e35960\n",
      "18:28:29 Searching Annoy index using 1 thread, search_k = 3000\n",
      "18:28:30 Annoy recall = 100%\n",
      "18:28:31 Commencing smooth kNN distance calibration using 1 thread\n",
      "18:28:33 Initializing from normalized Laplacian + noise\n",
      "18:28:33 Commencing optimization for 500 epochs, with 359746 positive edges\n",
      "18:28:51 Optimization finished\n"
     ]
    }
   ],
   "source": [
    "LS_one_mor <- RunUMAP(LS_one_mor, reduction='pca', dims=1:30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "18:28:51 UMAP embedding parameters a = 0.9922 b = 1.112\n",
      "18:28:51 Read 9539 rows and found 30 numeric columns\n",
      "18:28:51 Using Annoy for neighbor search, n_neighbors = 30\n",
      "18:28:51 Building Annoy index with metric = cosine, n_trees = 50\n",
      "0%   10   20   30   40   50   60   70   80   90   100%\n",
      "[----|----|----|----|----|----|----|----|----|----|\n",
      "**************************************************|\n",
      "18:28:53 Writing NN index file to temp file C:\\Users\\STUBER~1\\AppData\\Local\\Temp\\RtmpoF8MPH\\file3b006cb935d6\n",
      "18:28:53 Searching Annoy index using 1 thread, search_k = 3000\n",
      "18:28:55 Annoy recall = 100%\n",
      "18:28:55 Commencing smooth kNN distance calibration using 1 thread\n",
      "18:28:57 Initializing from normalized Laplacian + noise\n",
      "18:28:57 Commencing optimization for 500 epochs, with 409678 positive edges\n",
      "18:29:18 Optimization finished\n"
     ]
    }
   ],
   "source": [
    "LS_nal_no_mor <- RunUMAP(LS_nal_no_mor, reduction='pca', dims=1:30)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Load in files if necessary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_sal<-readRDS(file = \"/Doublet_processing/LS_sal.rds\")\n",
    "LS_mor<-readRDS(file = \"/Doublet_processing/LS_mor.rds\")\n",
    "LS_nal<-readRDS(file = \"/Doublet_processing/LS_nal.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts<-as.matrix(LS_sal@assays$RNA@data)\n",
    "write.table(data.frame(\"GENE\"=rownames(counts),counts),file=\"/LS_sal/counts.txt\",row.names=FALSE,sep=\"\\t\")\n",
    "markers <- FindAllMarkers(object = LS_sal, only.pos = TRUE, min.pct = 0.25)\n",
    "top_50<-markers %>% group_by(cluster) %>% top_n(50)\n",
    "write.table(data.frame(\"test\"=as.character(rownames(top_50)),top_50),file=\"/LS_sal/Top50Genes.txt\",row.names=FALSE,col.names=c(\"\",colnames(top_50)),sep=\"\\t\",eol=\"\\n\")\n",
    "cluster<-Idents(object=LS_sal)\n",
    "cluster<-as.matrix(cluster)\n",
    "cluster[,1]<-as.character(cluster[,1])\n",
    "cluster[,0]<-as.character(cluster[,0])\n",
    "cluster<-data.frame(\"x\"=rownames(cluster),cluster)\n",
    "write.table(cluster,file=\"/LS_sal/Cluster.txt\",row.names=FALSE,col.names=c(\"\",\"x\"),sep=\"\\t\",eol=\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"DoubletDecon\")\n",
    "location=\"/Doublet_processing/LS_sal/\" #Update as needed \n",
    "expressionFile=paste0(location, \"counts.txt\")\n",
    "genesFile=paste0(location, \"Top50Genes.txt\")\n",
    "clustersFile=paste0(location, \"Cluster.txt\")\n",
    "\n",
    "newFiles=Seurat_Pre_Process(expressionFile, genesFile, clustersFile)\n",
    "\n",
    "results=Main_Doublet_Decon(rawDataFile=newFiles$newExpressionFile, \n",
    "                           groupsFile=newFiles$newGroupsFile, \n",
    "                           filename=\"cntl\", \n",
    "                           location=location,\n",
    "                           fullDataFile=NULL, \n",
    "                           removeCC=FALSE, \n",
    "                           species=\"mmu\", \n",
    "                           rhop=1.1, \n",
    "                           write=TRUE, \n",
    "                           PMF=TRUE, \n",
    "                           useFull=FALSE, \n",
    "                           heatmap=FALSE,\n",
    "                           centroids=TRUE,\n",
    "                           num_doubs=100, \n",
    "                           only50=TRUE,\n",
    "                           min_uniq=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts<-as.matrix(LS_mor@assays$RNA@data)\n",
    "write.table(data.frame(\"GENE\"=rownames(counts),counts),file=\"/Doublet_processing/LS_mor/counts.txt\",row.names=FALSE,sep=\"\\t\")\n",
    "markers <- FindAllMarkers(object = LS_mor, only.pos = TRUE, min.pct = 0.25)\n",
    "top_50<-markers %>% group_by(cluster) %>% top_n(50)\n",
    "write.table(data.frame(\"test\"=as.character(rownames(top_50)),top_50),file=\"/Doublet_processing/LS_mor/Top50Genes.txt\",row.names=FALSE,col.names=c(\"\",colnames(top_50)),sep=\"\\t\",eol=\"\\n\")\n",
    "cluster<-Idents(object=LS_mor)\n",
    "cluster<-as.matrix(cluster)\n",
    "cluster[,1]<-as.character(cluster[,1])\n",
    "cluster[,0]<-as.character(cluster[,0])\n",
    "cluster<-data.frame(\"x\"=rownames(cluster),cluster)\n",
    "write.table(cluster,file=\"/Doublet_processing/LS_mor/Cluster.txt\",row.names=FALSE,col.names=c(\"\",\"x\"),sep=\"\\t\",eol=\"\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"DoubletDecon\")\n",
    "location=\"/Doublet_processing/LS_mor/\" #Update as needed \n",
    "expressionFile=paste0(location, \"counts.txt\")\n",
    "genesFile=paste0(location, \"Top50Genes.txt\")\n",
    "clustersFile=paste0(location, \"Cluster.txt\")\n",
    "\n",
    "newFiles=Seurat_Pre_Process(expressionFile, genesFile, clustersFile)\n",
    "\n",
    "results=Main_Doublet_Decon(rawDataFile=newFiles$newExpressionFile, \n",
    "                           groupsFile=newFiles$newGroupsFile, \n",
    "                           filename=\"cntl\", \n",
    "                           location=location,\n",
    "                           fullDataFile=NULL, \n",
    "                           removeCC=FALSE, \n",
    "                           species=\"mmu\", \n",
    "                           rhop=1.1, \n",
    "                           write=TRUE, \n",
    "                           PMF=TRUE, \n",
    "                           useFull=FALSE, \n",
    "                           heatmap=FALSE,\n",
    "                           centroids=TRUE,\n",
    "                           num_doubs=100, \n",
    "                           only50=TRUE,\n",
    "                           min_uniq=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts<-as.matrix(LS_nal@assays$RNA@data)\n",
    "write.table(data.frame(\"GENE\"=rownames(counts),counts),file=\"/Doublet_processing/LS_nal/counts.txt\",row.names=FALSE,sep=\"\\t\")\n",
    "markers <- FindAllMarkers(object = LS_nal, only.pos = TRUE, min.pct = 0.25)\n",
    "top_50<-markers %>% group_by(cluster) %>% top_n(50)\n",
    "write.table(data.frame(\"test\"=as.character(rownames(top_50)),top_50),file=\"/Doublet_processing/LS_nal/Top50Genes.txt\",row.names=FALSE,col.names=c(\"\",colnames(top_50)),sep=\"\\t\",eol=\"\\n\")\n",
    "cluster<-Idents(object=LS_nal)\n",
    "cluster<-as.matrix(cluster)\n",
    "cluster[,1]<-as.character(cluster[,1])\n",
    "cluster[,0]<-as.character(cluster[,0])\n",
    "cluster<-data.frame(\"x\"=rownames(cluster),cluster)\n",
    "write.table(cluster,file=\"/Doublet_processing/LS_nal/Cluster.txt\",row.names=FALSE,col.names=c(\"\",\"x\"),sep=\"\\t\",eol=\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"DoubletDecon\")\n",
    "location=\"/Doublet_processing/LS_nal/\" #Update as needed \n",
    "expressionFile=paste0(location, \"counts.txt\")\n",
    "genesFile=paste0(location, \"Top50Genes.txt\")\n",
    "clustersFile=paste0(location, \"Cluster.txt\")\n",
    "\n",
    "newFiles=Seurat_Pre_Process(expressionFile, genesFile, clustersFile)\n",
    "\n",
    "results=Main_Doublet_Decon(rawDataFile=newFiles$newExpressionFile, \n",
    "                           groupsFile=newFiles$newGroupsFile, \n",
    "                           filename=\"cntl\", \n",
    "                           location=location,\n",
    "                           fullDataFile=NULL, \n",
    "                           removeCC=FALSE, \n",
    "                           species=\"mmu\", \n",
    "                           rhop=1.1, \n",
    "                           write=TRUE, \n",
    "                           PMF=TRUE, \n",
    "                           useFull=FALSE, \n",
    "                           heatmap=FALSE,\n",
    "                           centroids=TRUE,\n",
    "                           num_doubs=100, \n",
    "                           only50=TRUE,\n",
    "                           min_uniq=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Calculating cluster 0\n",
      "Calculating cluster 1\n",
      "Calculating cluster 2\n",
      "Calculating cluster 3\n",
      "Calculating cluster 4\n",
      "Calculating cluster 5\n",
      "Calculating cluster 6\n",
      "Calculating cluster 7\n",
      "Calculating cluster 8\n",
      "Calculating cluster 9\n",
      "Calculating cluster 10\n",
      "Calculating cluster 11\n",
      "Calculating cluster 12\n",
      "Calculating cluster 13\n",
      "Calculating cluster 14\n",
      "Calculating cluster 15\n",
      "Calculating cluster 16\n",
      "Calculating cluster 17\n",
      "Calculating cluster 18\n",
      "Calculating cluster 19\n",
      "Calculating cluster 20\n",
      "Calculating cluster 21\n",
      "Selecting by gene\n"
     ]
    }
   ],
   "source": [
    "counts<-as.matrix(LS_one_mor@assays$RNA@data)\n",
    "write.table(data.frame(\"GENE\"=rownames(counts),counts),file=\"/Doublet_processing/LS_one_mor/counts.txt\",row.names=FALSE,sep=\"\\t\")\n",
    "markers <- FindAllMarkers(object = LS_one_mor, only.pos = TRUE, min.pct = 0.25)\n",
    "top_50<-markers %>% group_by(cluster) %>% top_n(50)\n",
    "write.table(data.frame(\"test\"=as.character(rownames(top_50)),top_50),file=\"/Doublet_processing/LS_one_mor/Top50Genes.txt\",row.names=FALSE,col.names=c(\"\",colnames(top_50)),sep=\"\\t\",eol=\"\\n\")\n",
    "cluster<-Idents(object=LS_one_mor)\n",
    "cluster<-as.matrix(cluster)\n",
    "cluster[,1]<-as.character(cluster[,1])\n",
    "cluster[,0]<-as.character(cluster[,0])\n",
    "cluster<-data.frame(\"x\"=rownames(cluster),cluster)\n",
    "write.table(cluster,file=\"/Doublet_processing/LS_one_mor/Cluster.txt\",row.names=FALSE,col.names=c(\"\",\"x\"),sep=\"\\t\",eol=\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading packages...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"package 'limSolve' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'ggplot2' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'gplots' was built under R version 3.6.3\"Warning message in library(package, lib.loc = lib.loc, character.only = TRUE, logical.return = TRUE, :\n",
      "\"there is no package called 'clusterProfiler'\"Warning message:\n",
      "\"package 'GenomicFeatures' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'S4Vectors' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'IRanges' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'GenomeInfoDb' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'tidyr' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'R.utils' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'R.oo' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'R.methodsS3' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'foreach' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'doParallel' was built under R version 3.6.3\"Warning message:\n",
      "\"package 'iterators' was built under R version 3.6.3\""
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading data...\n",
      "Processing raw data...\n",
      "Combining similar clusters...\n",
      "Creating synthetic doublet profiles...\n",
      "svd calculated PCA\n",
      "Importance of component(s):\n",
      "                 PC1    PC2     PC3     PC4     PC5     PC6     PC7     PC8\n",
      "R2            0.4229 0.0429 0.01994 0.01651 0.01266 0.01167 0.01043 0.00939\n",
      "Cumulative R2 0.4229 0.4658 0.48571 0.50222 0.51488 0.52655 0.53698 0.54637\n",
      "                  PC9    PC10    PC11    PC12    PC13\n",
      "R2            0.00875 0.00791 0.00705 0.00535 0.00513\n",
      "Cumulative R2 0.55512 0.56303 0.57008 0.57543 0.58056\n",
      "\n",
      " Attention: the number of pure cell types = 13  defined in the signature matrix;\n",
      "\n",
      " PCA results indicate that the number of cell types in the mixtures = 1 \n",
      "Step 1: Removing possible doublets...\n",
      "svd calculated PCA\n",
      "Importance of component(s):\n",
      "                 PC1     PC2     PC3     PC4     PC5    PC6     PC7     PC8\n",
      "R2            0.3759 0.05711 0.03498 0.01571 0.01077 0.0084 0.00784 0.00686\n",
      "Cumulative R2 0.3759 0.43303 0.46801 0.48372 0.49449 0.5029 0.51073 0.51759\n",
      "                  PC9    PC10    PC11    PC12    PC13\n",
      "R2            0.00591 0.00579 0.00558 0.00505 0.00435\n",
      "Cumulative R2 0.52350 0.52929 0.53487 0.53992 0.54427\n",
      "\n",
      " Attention: the number of pure cell types = 13  defined in the signature matrix;\n",
      "\n",
      " PCA results indicate that the number of cell types in the mixtures = 1 \n",
      "Step 2: Re-clustering possible doublets...\n",
      "Step 3: Rescuing cells with unique gene expression...\n",
      "Finished!\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAdVBMVEUAAAAA//9NTU1oaGh8\nZQB8fHyMjIyajACampqniQCnp6eysrK9mwC9rAC9vb3Hx8fQ0NDZ2dnhzQDh4eHpvwDp6enw\n8PD/AAD/FwD/LgD/RgD/XQD/dAD/iwD/ogD/uQD/0QD/6AD//wD//yr//4D//9X///9ydLIQ\nAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO2dC5uyvHpGs7W6p07rOLW6e+7+2m7+\n/0+sgCBiEgk+Oa91Xe87TIQEyL2GM6oGAD5GxZ4BgBIII9LPTqndad60Mjd+/+w22ebic8YA\nZAgh0mWrOg6zpt+K9IVHkAkhRLp7pNTzNumtSD+3/399zxyABAFEuvmwOTfX227atv31sFXb\nY9d0L9L091M/Sv/Z+aHeYaM2h6540/2+sTgIEIEAgdz325Xr9rvdTdt126ZdM4j09LtS+/ts\nKXW92fLT/7bpPro5dOjVOr3sJgLEJYBIT7twR6W+2mOf4738+fdDcx2n2Y3T3fT5bkc8Npd2\n7O7YiUMnSIrQIu1aBy7dXl5X/vz7ZTJNy3f3y7aroNta3bZu1+Y6brgAEiGASE8HNOouhZr+\nNx0cR1On3d2s4VyF6nbqju3GaX4qHSAu4Y6RmsP+7CLSqT3bsL//Moh02zptb/82/ucawIUA\nIp2mZ+22s127+e/DbHXD93ML28kHt63RoTuiAkiJEKeRd5PrSN+zkw3z34fZ6oc33banO8/w\nO908XQPMNYADQa7H7CZ3NlhPfz9mqx/+6c839OP0NzkcXm+RAIhOmAubp9tx0v5+guB7dkF2\n/nszHb6fbzjc9u6++lN6t91AdQ4y0wDLye4Ogasa7n4ASIfsRPoa73cASIfMRGqPldggQXpk\nJtJGbb44ZQfpkZlIAGmCSAACIBKAAIgEIEA4kR7XW4ehl1tVb1y+OSkHGZKaSLYXOQAkS3SR\njGMBZIR0bH/bV9H17ylRl6/7/aXnrdperFuk4/Y22XF49KiZvBKlueyGiYe3o7y2cd6pzbfw\nkgA4ICzSb2/CVxfyzf2W7/Y1Jt0vQ5svIn0Pt4cPIj3uCe8m3g6jts9SvLax7wowCeIhLNJW\n/Tbj5uPQvYmr0+TQPf4wtPlgYtNFTTZQj6eUDo+JVf92lFkbv93/J8VjsxAR+SOSn90Q8uvd\ni+GNJWaRbtuV+50/40NKw3Oz28fEanw7yryNa8PBFURFOn331xNrXsZgE+nY7aCdm+dx9fXo\n20AkiIpw+k7tA3gLRHr87H85f7VqnBaJpG0DkSAqwul7PuxpxnfXzXftZiO39MdT6v6uIMOu\nnbENRIKoyIt0+ZmHXHOy4fFzPPqZbmS+ZycbvmYivbaBSBAV4fS1urSnpC9P8X57+rs/f92e\nwG7PInyZT3+b20AkiMny9E1OEFgqOqjNoXtD9zTel63avbkg215jbS8EXXb9a7e+pxdk99Ot\nlaENRIKYLE6f0gw9ff6XO/8y8K8D/zby7wP/MfCfA/818t8Dfx3444+bJn+0/M/A/w7838Df\nRjAKgpO8SDeF/vmPf1Tqz4gECZO8SH9/36H8J0SChElepL/+w9/dNPpz7xEiQaLInWzweow0\nUKpIP1/ticn999MXRPlpqu/E/f23va1PYTliq1D9aQCRXPnZjH+l9vcit3Sf929HGXn+c2j9\n4wjLkRNp2CIhkisHNaH/qgAnkc57FxPu7TzepI5IEsjv2iGSI08e9Veh3URyM+HeTP/e5x9E\nEkL+ZAMiuXFqg7z9uTbNtd/F6761w7tI/VfjHBBJCESKzXbcCjX9rVHda5T8irQb3qC+7b89\nPq81liQfijTZJUGkVbQP+D6e7T1vD6fuCcch3Y+UP4bOh06+w3ksf2xVftrTcPvh+zq64u+N\n2k4ew2/L2kf722au9+Fhjdmmbp902R6ujXH0umGLFJk2x5q3TVhEehxTfQ/lg0jn4fTf5nec\n5vg4InqU7fsvyG6/T/Q4Vmyd+t7q5mIcvW442RCZ9o+65hsIzSKdJnsBv81Tv1xmH42f7Wc1\nH4+9hd/9YF+xderj8MvGOHrdcPo7MoY/TWaRtv0m4joeTj3GaT86XvvUj0da6vRa8/ncH5ft\n+sFHxaap233A9rUal70yjl43iBQZZ5GGn9fhcOppW9U/evJzN6D9ZP7N1Z1Ik2dPBpGsUx/H\nzdrOOHrduIbOOD4ircNZpO5Bx5+rZtz2OeLrWPR1/2S+29WLtG//O7d2DCJZp96Pgyfj6HXj\ndI71jv5zRFqFs0j3o/7d6WXcjZqyvX9y0dR8bjcxx/6/QSTr1I/nm6/G0evG4WSDfXxEWsdO\nTU82XBec/h5Plh2b50/UM43e0l6k8/2tGOdRJOvUr4Ovo9eNy5U86/iItI52A/M4/f2jZhdk\nNSI1h1Gl52MkF5Hab+Pt/iGSDC5roHtvgvFDRFpFe8zxuCC7VZOrQ9Ofz1L8Hvp3ZH49fbJ5\nCbRFpK/2GwnaCqa7dsapH4NX4+h147YyFCKJ0yZyfx/ujn+6w5IXkc4zKa6PC6nDwOOMwIBF\npJ/uuwd+HhVbp3492fA6et04hs78VwiRVtLdgL1tj4x+u4s0+650upXp8nq4l1xPh+3sSYth\n4Ec9X3o1i3QZrqcOA++mfny4NY5eN1xHis7+6WBjdtzTfrg7N9fv4UikPTmxu6l1bc16PHJx\naa73jdu53Vrdz1mYRGp/boZ9ymmBaerxguxOGUevG0SKz5NJ992lIa7jDUHDrTnn6cjnyfRf\n47dT9Rwau0hf6n6MNRRYpx5vEdoZR68bREqA43hRZjtc9RlDPNyiehoDPI68uV9K6mXbPX32\nfMpiylDW7pv9PI1knXq4fGUevW4QKQl+9rdUbr5eL7LeNNnNH1/oRlb773F/6rcb5ekz8wPr\nQ1l7bHSZjWSbuj2C2x0to9cNIoEbivsYdCASLGE8odAeo+3jzkuSIBIs4f7wRrcXqXgm9hVE\ngiWMZ+0U33qtBZFgEY8H3DecW9CASLCM86Hdq9vwrhM9iAQgwIehm+w5IxJUDFskAAEQCUAA\nRAIQ4G3o1JKRGkSCunkTuvuJhCUVIRJUzKIt0qKKEAkqhmMkAAHeh45dO4C3sGsHIAAiAQjw\nXiR27QDesjx04011ho8RCSpmceiUZujpc0SCilm8a4dIAGYWhQ6RAOwgEoAAi3btxgFONgBo\n4RYhAAEQCUCARffaDT/ZtQPQs/jBPk42AJhBJAABFu/a6UXidVwALdwiBCAAN60CCPAudEtf\nIoRIUDXv3iK0ZKR+HESCilku0uNeIf2YiAQVs1iku0zGSRAJaubtMZJ6Pv2tEAngFefT3wqR\nAF5wv45kOP+NSFAzLteRrJMgEtQMj1EACIBI75esSmTXYfkg0vslk60uD6pc6E9ApPdLJltd\nHlS50J+ASO+XTLa6PKhyoT8Bkd4vmWx1eVDlQn8CIr1fMtnq8qDKhf4ERHq/ZLLV5UGVC/0J\niPR+yWSrszYFr4Rb/Z+ASO+XTLa6RJrKhkzWCSK9XzLZ6hJpKhsyWSeI9H7JZKtLpKlsyGSd\nfDibk11ZRMqqqWzIZJ2wRXq/ZLLVJdJUNmSyThDp/ZLJVpdIU9mQyTpBpPdLJltdIk1lQybr\nBJHeL5lsdYk0lQ2ZrBNEer9kstUl0lQ2ZLJOEOn9kslWl0hT2ZDJOkGk90smW10iTWVDJusE\nkd4vmWx1iTSVDZmsk3JE+tMDqWW6L5lsdYk0lQ2ZrJOCRGKLVCSZrBNEer9kstUl0lRiZP+A\nBSK9XzLZ6hJpKjHWLHlSawuR3i+ZbHWJNJUYiDRWhEhZNZUYiDRWhEhZNZUYiDRWhEhZNZUY\niDRWhEhZNZUYiDRWhEhZNZUYiDRWhEhZNZUYiDRWhEhZNZUYFYn05noyIuXVVGLUI5LSDD19\njkhZNZUYiDSWIlJWTSVG7SJp7iT8Uyx83cyISAGoXaRFFdnaCP3ZChApAPWI9PZkw7o2EClS\nU4lRkUjrK0KkBJtKDERaUBEiJdhUYiDSgooSF2nVw5l+HtRMKhpBQaQFFaUuktvo4tP7qSoz\nEGlBRYgUoarMQKQFFSFShKoyA5EWVIRIEarKDETKH0RKAETKH0RKAETKH0RKAETKH0RKAETK\nH0RKAETKH0RKAETKn7AiJXErUho4Lmjaq6i43nEnsEiB2skA0wKZRFpfYwCK6x13ECkSiFQW\niBQJRCoLRIoEIpUFIkUCkTTV6M6YTAtdTqmsnW6cxuvoH06PSCOIpK9lXtW0UDvCgspcphun\nR6Q8QCR9LYmIpNgiZQIi6WuxieTS1otIq+bG2+gfTo9II4ikr8WPSBwjfT5loiCSvpZ3Ii1t\n6kkk5TCha0PrRv9wekQaQSR9LV5EcplwUUPyN7iJi+Q6i+njtioQyVq4uCXPIrlNE0UkiWKZ\nWjKo3HEUXwQTaXlDiCRRnFLWEcmpFqtIDu0gkkRxSllHpOXV6HaEh0I17PY7VqZMFb+bfsWH\niJRr5Y6j+CJi03FApMIqdxzFF4hkLUak5Ct3HMUXiGQtRqTkK3ccxReIZC1GpOQrdxzFF4hk\nLUak5Ct3HMUXiGQtRqTkK3ccxReIZC1GpOQrdxzFF4hkLUak5Ct3HMUXiGQtRqTkK3ccxReI\nZC1GpOQrdxzFF4hkLUak5Ct3HMUXiGQtRqTkK3ccxReIZC02ju70BNuShlbPYgZZ91v56q4Q\nJaZIavYzyNzESIzMKBlnPaHKfYFI9pYRqbDKfRF1107NZwGR5NrMN+uItK7x9v/7Tq26l02K\n/LS5tBiRSqvcF3FPNqjJv7tFk0Evc4dIdVfuizREGn5BJLk28806Iq1q/b5/1+3ITUXydA4T\nkequ3BdxRWqGN94/DpCetkheWnQpRqTSKvdFGiKZjpG8tOhSjEilVe6LyCJNN0gPkRTHSJ+3\nmW/WEWl9++3h0GiRGn76mDlEqrtyX8QWKTiIVHflvkAkazEilVa5LxDJWoxIpVXuC0SyFiNS\naZX7ApGsxYhUWuW+QCRrMSKVVrkvEMlajEilVe4LRLIWI1JplfsCkazFiFRa5b5AJGsxIpVW\nuS8QyVqMSKVV7gtEshYjUmmV+wKRrMWIVFrlvkAkazEilVa5LxDJWoxIpVXuC0SyFiNSaZX7\nApGsxYhUWuW+QCRrMSKVVrkvEMlajEilVe4LRLIWI1JplfsCkazFiFRa5b5AJGsxIpVWuS8Q\nyVqMSKVV7gtEshYjUmmV+wKRrMWIVFrlvkAkazEilVa5LxDJWoxIpVXuC0SyFiNSaZX7ApGs\nxUKdasJjmxlnHZEAagWRAARAJAABEAlAgDpE+gukwt8iECJiiARBQaSciZ0eGEGknImdHhhB\npJyJnR4YQaSciZ0eGEGknImdHhhBpJyJnR4YSUck2+2O7iASBCUhkUQjhkgQFETKmdjpgZFk\nRBJOfpoiGR/gcWSoL3Z6YCQdkSSPkJIVSbi+2OmBkXREkg0aIkFQkhFJOGmIBEFBpJAgUrEg\nUkgQqViSEUnJJg2RICjJiNSFjJMNbsROD4ykI1LD6W9nYqcHRhISSRREgqAgUkgQqVgQKSSI\nVCyItL6JD26SEyJ2emDETQGRSooRKcgkVoJGBWwgUsgmEKlYEClkE4hULIgUsglEKhZECtkE\nIhULIoVsApGKBZFCNoFIxZKQSLnda4dI8CAdkbK7+xuR4EEyIuX3zgZEggfJiCQctERF4nVc\npZKUSIIHSWmKJE3QqICNhESq4GSDNEGjAjYSEqkpf9dOmqBRARtpiZTVy08QCR4gUspNvCNo\nVMBGMiJFfh1XAk/prSBoVMBGOiLFvSCbwEWhFQSNCthIRqTIZ+0QCT4iIZFEQSQICiKtG3/d\nJNIEjQrYQKR146+bRJqgUQEbiLRu/HWTSBM0KrACEb9MtYSIGCJBCiCSl0mkCRkJWAMieZlE\nmpCRgDUgkpdJpAkZCVgDInmZRJqQkYA1IJKXSaQJGQlYAyJ5mUSakJGANcQR6R5NiTurEQlS\nIIpI6vEoBSItImQkYA0xRFLTV3J9mlJEghSIIJJ6iiYiLSFkJGANEURqshMpHsMshIwErCG2\nSDkcIyVAyEjAGhApC0JGAtYQWaTPY41IkAJxRRJINSJBCkQVSSLUiAQpEFMkkUwjEqRARJFm\nZ3hXgkiQAnFEEkQrxpJLM1kRMhKwhkJFchw/eUJGAtaASFkQMhKwBkTKggh9bexs0OG4ct0I\nETFE8tXXIm1WgxeBBkJEDJF89bVIm9XgRaCBEBFDJF99LdJmNXgRaCBExBDJV1+LtFkNXgQa\nCBExRPLV1yJtVoMXgQZCRAyRfPW1SJvV4EWggRARQyRffS3SZjV4EWjAmIvH67g+jhgi+epr\nkTarwYtAA6ZYeH8dFyIJ9LVIm9XgRaABQyr8v44LkQT6WqTNavAi0IA+FMPruBBpOTH6WqTN\navAi0IAxF3mJ5P+tW2+f+YjR1yJtVoMXgQbsac5HJPdJpInR1yJtVoMXgQaMuUAkR2L0tUib\n1eBFoAFjLhDJkRh9LdJmNXgRaMCYC0RyJEZfi7RZDV4EGjDmApEcidHXIm1WgxeBBoy5yOuC\nLCLBW7wINGDMRV63CCESvMWLQAMhIoZIvvpapM1q8CLQQIiIIZKvvhZpsxq8CDQQImKI5Kuv\nRdqsBi8CDYSIGCL56muRNqvBi0ADISLmLNIHd7zFI0Zfi7RZHiJrqwCRpJoIi88IIJITiCTb\nRFh8RgCRnEAk2SbC4jMCiOQEIsk2ERafEUAkJxBJtomw+IwAIjmBSLJNhMVnBBDJCUSSbSIs\nPiOASE4kJVL/GIXIRRpE8tV7Im2WR0oiTewJ/hiFVBNh8RkBRHIiIZFUg0iO+IwAIjmRjkhq\nEs3wzyOtaSIewyz4jAAiOZGOSE1uIiWAzwggkhNJiiQQeETy1XsibZYHIuWMzwggkhOIlDM+\nI4BITiBSzviMACI5kaJIEnlHJF+9J9JmeSBSzviMACI5gUg54zMCiOREUiIJgki+ek+kzfJA\npJzxGQFEcgKRcsZnBBDJCUTKGT+ZAAsp/X0JETFEAi8gkqkwZ2L0XuUgkqkwZ2L0XuUgkqkw\nZ2L0XuUgkqkwZ2L0XuUgkqkwZ2L0XuUgkqkwZ2L0XuVkIJLoF6UgEnghfZFUI5l0RAIvJC+S\nmvwvACKBFxBJsnZbw9m9jgtcSF6kUnbtgm71YvRe5aQvUiEnGxCpbNIXiS2SOzF6r3KSF6mY\nY6QAbYzE6L3KQSTJ2l0b9kWM3qscRJKs3bVhX8TovcpJXiSOkVYQo/cqJ32ROGvnTozeq5wM\nRBIFkcALiGQqDNGwL2L0XuUgkqkwRMO+iNF7lYNIpsIQDfsiRu/VQkLCmAgRMUSCz0CkDkSC\nz0CkDkSCz0CkDkSCz0CkDkSCz0CkDkSCz0CkDkSCz8hZJO61cyR2T5ZMxiJx06orsXuyZPIV\nSbFFciV2T5ZMtiIpdu2cid2TJZOtSE0hIvFeuzJAJHNVbJFgMYhkrgqRYDGIZK4KkWAxiGSu\nCpFgMYhkrgqRYDGIZK4KkWAxiGSuCpFgMTmLJAgiwWcgUgciwWcgUgciwWcgUgciwUIyMMZA\niIghEiwEkWwgEiwEkWwgEiwEkWwgEiwEkWwgEiwEkWwgEiwEkWwgEiykQJHuIZT4DkxEgoWU\nJ9LdH9UIxBGRYCHFiXR/Hdfjv09AJFhIaSI9vY4LkRYRuyeLoDSRmkkIPz9I4nVcsJByReJk\nw1Ji92QRlCtSw67dQmL3ZBEULdLHeUQkWAgi2UAkWEipIj0uJn0CIsFCihWJC7KLid2TRVCq\nSJy1W07sniyCAkUSBJFgIYhkA5FgIYhkA5FgIYhkA5FgIYhkA5FgTr7GGAgRMUSCOYi0AkSC\nOYi0AkSCOYi0AkSCOYi0AkSCOYi0AkSCOYi0AkSCOfWIxBOyjsTuybyoRqTHTd85v/wkQBsj\nsXsyL2oRSY0hVGyRlhG7J/OiEpEe77VTWe/a8TquVKlEpKYQkQK0MRK7J/OiNpF41HwxsXsy\nLyoTiXd/Lyd2T+ZFXSLJvEQIkeCFykSaHU+vBJFgTl0izYdWgkgwB5FWMLm2qzlt7BFEShZE\nWoF6GZCo1aXhEMTuybyoRyRBEAnmINIKEAnmINIKEAnmINIKEKleDMJkbIyBEBFDpHpBpNUc\nt01z2art71iCSPWCSGs5tVeINu2FotEkRKoXRFrLTv00Z7VtftRuKEKkekGktbQbpLM6PD2s\n/jLQIFIlINJaWn/26oRI0IJIa9mp80ltGnbtoAWRhkfNXe8uPbXnGb7bCU+zqhCpQqoXSa1+\n1Py4aY+Qmu3Po66XAfdaV4FI0aldJDV9ylzswT5Eqo7KRRreHSSTxHgi8Tqu2FQuUjOKtOIJ\nvNO+O3N3ea7qaaBhi1QJiNSLtOIYadf/mVab0SREqhdEWnuMdFS7ayvSUX09VTWvCJGqAJHW\nirRR1/uGjAuygEirRep26xAJehBprUjb+xapvXF1WtW8IkSqAkRae0H2fox02qjjc1UNIlUI\nIq29RajZ36+vjLfaIVLFINJ62utIav+4QwiRKgaRBEGkekEkQRCpXhBpPT/trt3X6VGASBVQ\nizAm5AO1u59s2I8liFQBiCTMQW3ajRGnvysDkYTZqHP3kwuydYFIwozXnbhFqCoQSZjDuEXa\nD0WIVAGIJM13d4z0u+HOhqpAJGE0D2IjUgUgkoG199ohUp0gkiGDa+/+1tX1MiBRq0vDIYjd\nk5FBJEMEVz+P9H2ZFyFSBSCSIYHrn5BVc5cQqQIQycDaXbvrz9fcpXgihWBoLHZPRgaRTCEc\ns+ge4N/v7dQltkgVgEgGPjzZcG6/tO84rapBpJJBJANrj5F6Tv094Lvn6RGpXBDJwAciXb9v\nm6Pt6Xqzaf80PSKVCyIZWC3Sb3uy4dDfcMcF2WpAJANrj5Ha0wzH61DJ5lFVg0glg0gGVt8i\ntD+9FL0MNIhUGogkSCvdVVP8MtAgUmkgkiCGrRciVQAiCYJI9YJIgiBSBdRujAHRLCFSBSCS\nFtEsGW7nRKSSQCQtollCpApAJC2iWWLXrgIQSYtolhCpAhBJi2iWEKkCEEmLaJYQqQIQSYsx\nF2vutUOkCkAkLcYMfviE7FNdLwMStbo0HILYPRkKRNJijOBnT8jOKpsPCFTq1HAIYvdkKBBJ\nizGBiORI7J4MBSJpMeaiCJFCMDQWuydDgUhazCF8pD1fkQK0MRK7J0OBSFqMuZicbECkJcTu\nyVAgkhZjLianvxFpCbF7MhSIpMWYC6UdtMNNqxWASFqMuVhzjIRIFYBIWoy54IKsI7F7MhSI\npMWYizW3CNmrQqQiQCQtolli164CEEmLaJYQqQIQSUuIiCFSSSCSlhARQ6QswRgXPCTqtG/3\n6vaP775EpCxBJBfkA7XrD4/UZjQJkbIEkVwQz9NR7a6tSEf1NRQhUpYgkgviedqoa3+bK2ft\nMgeRXBDPU7dbh0gFgEguiOdpe98indV2KEKkLEEkF8TzdD9GOm3UcShCpCxBJBfkA7W/39ew\nG0sQKUsQyQVTBIc7fFbctNpeR1L7n0llLwMNIqUPIrlgTqB6GvgERMoSRHLBEkDFy09ciN2T\n4iCSC5ZkrBHp8tWdYrhuj5PCeCKFYGgsdk+Kg0gu2GLoLtJlo/btz5Oa3CHEFilPEMkFczAe\nh0fL87hVX9du4Hf3uIyESHmCSC6Yg7FCpJP6Hof3ajxvh0hZgkguGHMxnrRzyOOXuo7Dl8eF\nJETKEkRywZ5BR5Gerjlxr13mIJIL9gg6irRBpIJAJBfeJNDtguyXOo3Dp/783UttumFfINIn\nIJIL+gA+rpA43SJ0fpz0vmw42ZA5iOSCbJgOavN9vv08f2+4aTV3EMkF4TR9jxf8vx6FiJQl\niOSCdJwuh/blJ/vvy6QMkbIEkVwIETFEShuMESBExBApbRBJgBARQ6S0QSQBQkQMkdIGkQQI\nETFEShtEEiBExBApbRBJgBARQ6S0QSQBQkQMkdIGkQQwRfBxj93naUSktEEkAcwJHJ6kQKRl\nxO7J9SCSAJYA3p+RRaRlxO7J9SCSAJZk3DdLWYgU5MVbeoZZiN2T60EkAd7nPw+RHEo9Ebsn\n14NIApiDkdXJBkT6AEQSwBwMRHIjdk+uB5EEMOZCm/51IFLaIJIAplgozdBaECltEEkAQyqU\ndnAliJQ2iCSAPhSyiUektEEkAfQBnF4hQaRlxO7J9SCSACEihkhpg0gChIgYIqUNIgkQImKI\nlDaIJECIiCFSGhiEwRgJQkQMkdIAkTwSImKIlAaI5JEQEUOkNEAkj4SIGCKlASJ5JETEECkN\nEMkjISKGSGmASB4JETFESgNE8oghFeOtdtO3EqwFkdIAkTyiD4UaUjgOfAIipQEiecQSQDUZ\n+AhESgNE8oglGWIPmyNSGiCSRyzJKEAk3ms3BZE8siCb+R4jBSV2T74HkTxiDgYiuRG7J9+D\nSB4x5kLOI0RKBETyiCkWgh4hUiIgkkcMqZD0yCCSn0P+eMTuyfcgkkf0oRD1SLg2exPxiN2T\n70Ekj2gzMf6Vl/lzj0hpgEgeCRExREoDRPJIiIghUhogkkdCRAyR0gCRPBIiYojkC5MaCBOc\nEBFDJF8gUjKEiBgi+QKRkiFExBDJF4iUDCEihki+QKRkCBExRPIFIiVDiIghki8QKRlCRAyR\nfIFIyWDMxeNtXNxrt4gYvYdIyWCKxfBaO4mIIZIvECkZDKlQQm/ieq4EkYRBpGTQh2J4pZ1M\nPhHJF4iUDMZc9CKJPHxatEhRX8eFSMlgD4jMZgmRfIFIyfA+IJ5eECkLIiFSXN4HBJGWtRyj\n9xApGd4HBJGWtRyj9xApGewBkTkHjki+QKRkeBMQ1XCyYWHLMXoPkZLhXUA4/b205Ri9h0jJ\nEDRriCQMIiVD0KwhkjCIlAxBs4ZIwiBSMgTNmkeR4jHMQozeQ6Rk8BbtacpfBgoCkaAlaNYQ\nSRhESoagWUMkYRApGYJmDZGEQaRkCJo1RBIGkZIhaNYQSRhESoagWUMkYRApGYJmDZGEQaRk\nMMVD4mbVl6whkjCIlAzmdCDSIhAJWizhEIs9IvkCkZJhSUTEsoZIwiBSMiyJiFjWShRpJEbv\nIVIymIPBMZIbMXoPkcIx3RwAAAtYSURBVJLBHAxEciNG7yFSMhhzIRh6RPIFIiWDKRaSmUck\nXyBSMhhSIRp5RPIFIiWDPhSyiUckXyBSMmgzMXsbwacgki8QKRlCRAyRfIFIyRAiYojkC0RK\nhhARQyRfIFIyhIhYHSK5hVqG2OlJlBgrMUTEEMkXMhkojhgrMUTEEMkXMhkojhgrMUTEEMkX\nMhkojhgrMUTEEMkXMhkojhgrMUTEEMkXMhkojhgrMUTEEMkXMhkojhgrMUTEEMkXMhkojhgr\n0RwMHuxzw48qITJQHDFWojEXvNfOET+qhMhAccRYiaZYKLZIjvhRJUQGiiPGSjSkQpW/ayf9\n1Zd+VAmRgeKIsRJNKatAJOH6/KgSIgPFEWMlmkOGSG74USVEBoojxko0ZwyR3PCjSogMFEeM\nlWiOGCK54UeVEBkojhgrURuxKt7ZgEjFEmMlhggaIvlCJgPFEWMlhggaIvlCJgPFEWMlhgga\nIvlCJgPFEWMlCqdJCyL5QiYDxRFjJQqnSQsi+UImA8URYyUKp0kLIvlCJgPFEWMlCqdJi7RI\nwjfJCSHSqY7IZKA4YqxE4TRpERcpoVoeiHSqIzIZKI4YK1E4TVoQyRcyGSiOGCtROE1aEMkX\nMhkojhgrUThNWhDJFzIZKI4YK1E4TVoQyRcyGSiOGCtROE1aEMkXMhkojhgrUThNWhDJFzIZ\nKI4YK1GfsbQfo0Ak2QwUR4yVGCJoiOQLmQwUR4yVGCJoiOQLmQwUR4yVGCJniOQLmQwUR4yV\nGCJniYokfMueSKc6IpOB4oixEv2GdVZXUiJJI9KpjshkoDhirERTLBDJFZFOdUQmA8URYyWa\nYoFIroh0qiMyGSiOGCvRFAtEckWkUx2RyUBxxFiJhlSIZhWRfCGTgeKIsRINqUAkZ0Q61RGZ\nDBRHjJVoSAUiOSPSqY7IZKA4YqzEEBFDJF/IZKA4YqzEEBFDJF/IZKA4YqzEEBFDJF/IZKA4\nYqzEEBGrQyS3bvrYIcEMgAAhIoZIryBSYYSIGCK9gkiFESJiiPQKIhVGiIgh0iuIVBghIoZI\nryBSYYSIGCK9gkiFESJiiPQKIhWGIRWi33qCSK8gUmHoQ6EaP28REn5NQlIgUtVoM6Em/wsg\nHXxEQqTk0GYCkVaASFWjzUQpIkntSi7az0SkqrEktQCRhBu2gkhVY06gh5foh64PkSAUpgje\nNEIkJxCpaizJQCQnEKlqLAFEJCcQqWrMCfRwQTZ0fYgEoTBF0MstQqHrQyQIRYiIIdIriFQY\nISKGSK8gUmGEiBgivYJIhREiYlWL5NWvoFEBGyEihkiIVDwhIoZIiFQ8ISKGSIhUPCEihkiI\nVDwhIoZIiFQ8ISKGSIhUPCEihkiIVDwhIoZIiFQ8pghy06oriFQ15gTyGIUbiFQ1lgDyYJ8T\niFQ1IQJY9eu4EKkODEnlGMkZRKoafQA5RnIHkarGEkCOkZxApKqxBBCRnECkqrEEEJGcQKSq\nsQQwXZGcz6iFAJGqxhDVJuWTDWk2jEhVY4pgyqe/02wYkaomRMQQCZGKJ0TEEAmRiidExBAJ\nkYonRMQQ6WMzECl1QkQMkRCpeEJEDJEQqXhCRAyREKl4QkQMkRCpeEJEDJEQqXhCRAyREKl4\nQkQMkRCpeIy5UC8Dq0EkRCoeYwbVfGA9iIRIxWOMoJoNfAAiIVLxGBOongc+IZ5IIRgaQ6Sq\nMYfwZWB9nj+uIQcQqWqMuUAkRxCpaoy5QCRHEKlqjLlAJEcQqWqMuUAkRxCpaoy5QCRHEKlq\njLlAJEcQqWqMuUAkRxCpakJEDJEQqXhCRAyREKl4QkQMkRCpeEJErA6RADyDSAACIBKAAIgE\nIAAiAQiASAACIBKAAIgEIAAiAQiASAACIBKAAIgEIAAiAQiASAACIBKAAIgEIAAiAQiASAAC\nIBKAAIgEIAAiAQhQuUjt4k+/kWwsn31T2Tjym7rmlWg+UWMDi0Y3tqabQ3O5UtMWnmZGV4tt\nPkyja0bUjG2e1jjn5tWir9y4QD6pWyR1/zdfDerpw7FU3z+6rwgcp3+tXT3/9250y5w/D9jm\nvP92R22xZlZMC9Qn9LUW8yxqxjaG3LBEprWonRXTug1AHSIZojHpCM1G46WfVKP/+mv9X0xD\nJaZGTW2a5/x10Dbn45/r97NiWKBJphfPorZN20rUzWKjXyDdrJhn3jd1iGRYTHeR9PsNrz06\n+VU5iaQdXT/nuhFkRNIvkKkW2yzq29Tu2OrrM6wW1wXyTyUi6ZdzjUjdwDwG6mW7Mx3bSaT5\n6OY5130uJJJ2gYwiWWbRIFLfhMnGBasFkaJhN0knwUuWbOtqviFpniLzVKyUPY0vlhra036q\nn/N+9rTF2uVvtAvkPoumwxjD6LoxTGtRW2xatwGoRiQ9tnNfzaI/vI8PLbl7rV1Xuqid6Rwu\nnvNhE6M7JaatpdEu0PjXfllqhM7a6ac0zYp+8f1TuUiCRDjl6pfiFsgrrKwe03owlIsUl0eM\n5U9knVfTxzqUZshSLlJsmRv9lUe34ghILX/qbdqJ3g1RUbOfb8pFim3zYrh+u7w4CjLLn36b\nVuJ3Qwgs1+qH/+YTaMsliu2XWOdHym7Fxsolip3XorbYsXKRNkNQh0i2c0WGzwzlAsWm9saf\nGmMWFns+pnNei5Yzf5/Oi2tX+KYSkaz94VQuUGwv1Fx5XFwsM3/rTFpcLGiS4wQ+qUUkS/eF\n3iK9NenlYGh5saly3yY5Lb+QSWyREiLCMZJ5Xl4GVhRHIcbxCsdIKaFmP9+UixSXR4zlT26d\nF9/Lc7RHGsYjdv3u1CfFpnmRLg5X+erl/2DO169zX9QhktdOdSz2ammMyrVjGItl5tytzRDU\nIVJSu1le9xtjVO6GzJwnR+rzJ0W0g1ANXs9kxKjcDZk5T43kZ1CMSKdFtfg8tx6lcjdk5jwt\nMphFKVJa1DJOM6wm8eOdNeQwjzKk9HeNLZJLeUo9ZySDWRQhpT1tjpE4RsqVlM79cNbOpfKU\nes5G6vMng9IMxcIwLyLFUSp3Q2bO0yPx2ZMnpQXmlIPEvKRB2nMHkAmIBCAAIgEIgEgAAiAS\ngACIBCAAIgEIgEgAAiASgACIBCAAIgEIgEgAAiASgACIBCAAIgEIgEgAAiASgACIBCAAIgEI\ngEgAAiASgACIBCAAIgEIgEgAAiASgACIBCAAIgEIgEgAAiBSrpy/NurrNC9tv0ho/DIhNSHw\n3FUHKzhTDr0f28tzMSJFghWcJ99qc9saXW8/nk16EmksAe+wlrPkMgj0pb6ePkCkSLCWs+Sg\nvvuB6/7Y/jhu1aYbMIl0Vdtm+HkrOajNof94nLI57ZTavRx0wTIQKUt26jz9dd8dBu0ayxZp\nr37bHz83A5X6HkafTHnsj6WOQea/PBApS55dOandtbnu1Mki0qnfB/xSl1vJ5tycN+rnacpN\n6+bPfcMFriBSljy7slfXpt1r29uOkbbdSK0oqvXm5tB+NiW7dR+ASFkyd2U8x20W6dgeVv22\n/91L+lHHKQ9K7c9PO4zgACJlyX48RjpdF4p0VZv2rPnFJFLzvbn9nJ1Nh6UgUpZ8D2ftfvt9\ntfEDy+nvw23fbbttZiJNRz0dthwjrQSRsmS8jrRrT7PtH4c3FpHOanfu/FPdCbzu7MN+fmDE\nVaeVsN7y5Ku7s+Gyb/fXmp/2NNztIGhvvyC7VZt2z248a3d6mnLbnsXjrN1aEClTdtN77fpf\nNhe7SCfVa6JUN/7+ecqfvsLfgMtQEoiUKz/7mxA/91+OW6W++q2NWaSr6o+sbiV7tT3Opuzv\nbMCjlSBSPdy2SJ0xHAd5gHVaD7v7/T+I5AHWaS0Md9chkhdYp7Ww6c8uNIjkBdYpgACIBCAA\nIgEIgEgAAvw/ny0xmGNxvEUAAAAASUVORK5CYII=",
      "text/plain": [
       "Plot with title \"Color Key\n",
       "and Histogram\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(\"DoubletDecon\")\n",
    "location=LS_one_mor/\" #Update as needed \n",
    "expressionFile=paste0(location, \"counts.txt\")\n",
    "genesFile=paste0(location, \"Top50Genes.txt\")\n",
    "clustersFile=paste0(location, \"Cluster.txt\")\n",
    "\n",
    "newFiles=Seurat_Pre_Process(expressionFile, genesFile, clustersFile)\n",
    "\n",
    "results=Main_Doublet_Decon(rawDataFile=newFiles$newExpressionFile, \n",
    "                           groupsFile=newFiles$newGroupsFile, \n",
    "                           filename=\"cntl\", \n",
    "                           location=location,\n",
    "                           fullDataFile=NULL, \n",
    "                           removeCC=FALSE, \n",
    "                           species=\"mmu\", \n",
    "                           rhop=1.1, \n",
    "                           write=TRUE, \n",
    "                           PMF=TRUE, \n",
    "                           useFull=FALSE, \n",
    "                           heatmap=FALSE,\n",
    "                           centroids=TRUE,\n",
    "                           num_doubs=100, \n",
    "                           only50=TRUE,\n",
    "                           min_uniq=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Calculating cluster 0\n",
      "Calculating cluster 1\n",
      "Calculating cluster 2\n",
      "Calculating cluster 3\n",
      "Calculating cluster 4\n",
      "Calculating cluster 5\n",
      "Calculating cluster 6\n",
      "Calculating cluster 7\n",
      "Calculating cluster 8\n",
      "Calculating cluster 9\n",
      "Calculating cluster 10\n",
      "Calculating cluster 11\n",
      "Calculating cluster 12\n",
      "Calculating cluster 13\n",
      "Calculating cluster 14\n",
      "Calculating cluster 15\n",
      "Calculating cluster 16\n",
      "Calculating cluster 17\n",
      "Calculating cluster 18\n",
      "Calculating cluster 19\n",
      "Calculating cluster 20\n",
      "Calculating cluster 21\n",
      "Calculating cluster 22\n",
      "Calculating cluster 23\n",
      "Selecting by gene\n"
     ]
    }
   ],
   "source": [
    "counts<-as.matrix(LS_nal_no_mor@assays$RNA@data)\n",
    "write.table(data.frame(\"GENE\"=rownames(counts),counts),file=\"/Doublet_processing/LS_nal_no_mor/counts.txt\",row.names=FALSE,sep=\"\\t\")\n",
    "markers <- FindAllMarkers(object = LS_nal_no_mor, only.pos = TRUE, min.pct = 0.25)\n",
    "top_50<-markers %>% group_by(cluster) %>% top_n(50)\n",
    "write.table(data.frame(\"test\"=as.character(rownames(top_50)),top_50),file=\"/Doublet_processing/LS_nal_no_mor/Top50Genes.txt\",row.names=FALSE,col.names=c(\"\",colnames(top_50)),sep=\"\\t\",eol=\"\\n\")\n",
    "cluster<-Idents(object=LS_nal_no_mor)\n",
    "cluster<-as.matrix(cluster)\n",
    "cluster[,1]<-as.character(cluster[,1])\n",
    "cluster[,0]<-as.character(cluster[,0])\n",
    "cluster<-data.frame(\"x\"=rownames(cluster),cluster)\n",
    "write.table(cluster,file=\"/Doublet_processing/LS_nal_no_mor/Cluster.txt\",row.names=FALSE,col.names=c(\"\",\"x\"),sep=\"\\t\",eol=\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading packages...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in library(package, lib.loc = lib.loc, character.only = TRUE, logical.return = TRUE, :\n",
      "\"there is no package called 'clusterProfiler'\""
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading data...\n",
      "Processing raw data...\n",
      "Combining similar clusters...\n",
      "Creating synthetic doublet profiles...\n",
      "svd calculated PCA\n",
      "Importance of component(s):\n",
      "                 PC1     PC2     PC3     PC4     PC5     PC6     PC7     PC8\n",
      "R2            0.3196 0.04234 0.02653 0.02282 0.01502 0.01331 0.01275 0.01093\n",
      "Cumulative R2 0.3196 0.36199 0.38852 0.41134 0.42636 0.43967 0.45242 0.46335\n",
      "                  PC9    PC10    PC11    PC12\n",
      "R2            0.01016 0.00776 0.00725 0.00619\n",
      "Cumulative R2 0.47351 0.48127 0.48852 0.49471\n",
      "\n",
      " Attention: the number of pure cell types = 12  defined in the signature matrix;\n",
      "\n",
      " PCA results indicate that the number of cell types in the mixtures = 1 \n",
      "Step 1: Removing possible doublets...\n",
      "svd calculated PCA\n",
      "Importance of component(s):\n",
      "                 PC1     PC2     PC3     PC4    PC5     PC6     PC7     PC8\n",
      "R2            0.2909 0.04934 0.03484 0.02441 0.0102 0.00923 0.00658 0.00597\n",
      "Cumulative R2 0.2909 0.34027 0.37511 0.39952 0.4097 0.41895 0.42553 0.43150\n",
      "                  PC9    PC10    PC11    PC12\n",
      "R2            0.00549 0.00542 0.00491 0.00462\n",
      "Cumulative R2 0.43699 0.44241 0.44732 0.45194\n",
      "\n",
      " Attention: the number of pure cell types = 12  defined in the signature matrix;\n",
      "\n",
      " PCA results indicate that the number of cell types in the mixtures = 1 \n",
      "Step 2: Re-clustering possible doublets...\n",
      "Step 3: Rescuing cells with unique gene expression...\n",
      "Finished!\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAdVBMVEUAAAAA//9NTU1oaGh8\nZQB8fHyMjIyajACampqniQCnp6eysrK9mwC9rAC9vb3Hx8fQ0NDZ2dnhzQDh4eHpvwDp6enw\n8PD/AAD/FwD/LgD/RgD/XQD/dAD/iwD/ogD/uQD/0QD/6AD//wD//yr//4D//9X///9ydLIQ\nAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO2dDXvqsHZmdaHQNOkknEnD7cd83Dtt\n+f8/cWKMwQbJliztLcla63lOQoRlGeldx7YwxlwAIBqTewMAtoCOSF8HYw6n56aNu/Hbc7/V\ndj+SGwaQBg2RfvbmyvtT04siveERVIKGSDePjJnukxZF+vr9+Ud64wBSoCDSrw+778v59zBt\n3/35vjf7z2vTvUjjv0/9Iv1z3w/13ndm934t3l3/3s04CJABhUAe+/3Kef/RHaYdrvumw2UQ\nafK3McfbZhlz/rXlq/9rd33q16H3Xq3Ty2EiQF4URJocwn0a89ad+3zeyqd/v1/O9zqHe71f\nfT66BT8vP93S13MnTp2gKLRFOnQO/FyP8q7l079/RnU6Pq5/7K8ruO6tfvdu58v5vuMCKAQF\nkSYnNOYmhRn/GD+8L2ZOh5tZw1yFuR7UfXY7p+epdIC86J0jXd6P3yEinbrZhuPtj0Gk373T\n/vffTn6rAUJQEOk0nrXbPx3aPf89bNb18W1uYT964ndv9H49owIoCY1p5MPofaSPp8mG57+H\nzeof7677nus8w5/x7umssNUAAai8H3MYXdkwO/392Kz+8Vc/39Av01/k8P56iQRAdnTe2Dz9\nnicdbxMEH09vyD7/fRk/vs03vP8e3b31U3q/h4HmW2WjAfyp7gqBsxmufgAoh+pEertf7wBQ\nDpWJ1J0rsUOC8qhMpJ3ZvTFlB+VRmUgAZYJIAAlAJIAEIBJAAvREerzfOjx6uVT1l58PJuWg\nQkoTae5GDgDFkl0k51IAFZE6tn+6W9H19ykxP2+360u/92b/M7tH+tz/VvscPnp0Gd0S5fJz\nGCoPd0d5beP7YHYfiV8JQACJRfrTm/B2Dfnudsl3dxuT6x9Dmy8ifQyXhw8iPa4Jv1beD4t2\nn6V4beN4LcAkyEdikfbmz+W++3i/3onrqsn79eMPQ5sPRjb9mNEO6vEppfdHZdPfHeWpjT/X\nnyfDx2YhI+nPSL4OQ8jPNy+GO5a4Rfrdr9yu/Ll/SGn43Oz+Udnc747y3Mb5wskVZCV1+m63\nJ7bcjGFOpM/rAdr3ZbqsfT32NhAJspI4fafuA3geIj1+9398v3VqnLxEsraBSJCVxOmbnvZc\n7veuez60e1q4oz+fMrd7BTkO7ZxtIBJkJb1IP1/PIbdMNjx+389+xjuZj6fJhrcnkV7bQCTI\nSuL0dbp0U9I/k3gvTn/389fdBHY3i/Dmnv52t4FIkJPU6Xs3u/frHbrH8f7Zm8PCG7Lde6zd\nG0E/h/62Wx/jN2SP472Vow1EgpxEpm80AffXG/868G8D/37nPwb+18D/Hvg/d/7vwN8G/v73\n3/X/veP/DfznwH8N/PcdjAJ1koVOSqRfhf7l7//DmH9EJCiY4kX6p9sO738iEhRM8SL97Z//\n4Vejf+w9QiQolPJFup4jDWxVpK+3bmLy+DH5giiZpvod/PH2122+VKaplkgn0l8GECmUr919\nyuZ4KwpL9/dxcZE7ZqKOQaQ0pN8jIVIo76O5z9tXBQSJ9H0MMeHWzuNO6oiUAkTKzsSj/l3o\nMJHCTLg109/3+QuREoFIuTl1Qd5/nS+Xc3+Id/3WDnGR+q/GeUekRPh34f2/TMfTiLSO/X0v\ndOkvjbreRklWpMNwB/V9/+3xdfVYkfiP1kIVRFpH9wHfx2d7v/fvp+snHId0P1L+ePT9fpXv\n/fte/vgP7qubhjsO39dxLf7Ymf3oY/hdWffR/q6Z8+3x0GNztbtPuuzfzxfn4m2DSJnpcmy5\n28SMSI9zqo+hfBDpe5j+2/251/l8nBE9yo79F2R33yf6eV/xbO1bq7sf5+Jtg0iZ6f5Tt3wD\noVukk3nw5zIR6efpqftzx6c1f372Fn70D/sVz9b+HP7YORdvG0TKzMMUa/GrSPt+F3G+n049\nlume+jz3qb+faZnT65q/v/vzskP/8LFiV+3uGLC7rcbP0TgXbxsmGzITLNLw+zycTk32Vf1H\nT75uBnTPPH9z9VWk0WdPBpFma3/ed2sH5+Jtw/R3ZoJFun7Q8etsWbb7HPH5XvR2e+b5sKsX\n6dj9+O7sGESarX28Pzw5F28bRMpMsEi3s/7D6WXZx4VG5na41f3+saz5u9vFfPY/BpFmaz8+\n33x2Lt42HNpl5mDGkw1nj+nv+2TZ52X6jJlysVvai/R9uyvG912k2dqvD18XbxsmGzLT7WAe\n099f5ukNWYtIl/e7StNzpBCRum/jvf5DpDQgUma6c47HG7J7M3p3aPx7KsWf9/4emW+TZ3Yv\ngZ4R6a37RoJuBeNDO2ftx8Ozc/G2iRRp9H8SIq2jS+Tx9vh6/nM9LXkR6ftJivPjjdThwWNG\nYGBGpK/rdw98PVY8W/t1suF18bZhj5Sb6wXY++7M6M/1TZrjtXS8l7nm9f1Wcj69758+aTE8\n+DLTt17dIv0M76cOD5ZqP57cOxdvGyYbsnOcnGw8nfd0Tx6+L+ePoeu7yYnDr1rnzqzHRy5+\nLufbzu2721vd5ixcInW/d8Mx5bjAVfv+huzBOBdvG6a/8zMx6Xa4NMT1fkHQcGnO93jh71H9\nt/u3U/W8X+ZFejO3c6yhYLb2/RKhg3PxtkGkAvi8vymzH971uYd4uET1dA/wfeHd7a2kXrbD\n5LnplMWYoaw7NvuaLDRbe3j7yr1424SGzrk8IsXwdfxN5e7t9U3WX00Ozx9fuC5sjh/346k/\n10Umz7k/sD6UdedGP08LzdXuzuAOnzOLt43/ZMMD+/OI1AaG6xhsBEw2zC+PSJvmPqHQnaMd\n825LkQSEzswuj0ib5vbhjetRpOEzsa+EhK47qEOkNvkcHdrzrdcWwkJnEKlVHh9w3zG3YCEw\ndO7rqxBp43y/d0d1O+51Yof3kQASgEgACUAkgAQgEkACEAkgAYuhMz4LXRAJ2mYhdHNX1z0t\niUjQMF57JK8VIRI0DOdIAAlYDt3sod3oCixEgobh0A4gAYgEkIBlkZi1A1iEyQaABCASQAI4\ntANIgFfoEAlgHkQCSIDXod39wcxxHiJBy/BtFAAJQCSABHhdazddEJEAnvH+YB8iAbjx/4Qs\nkw0ATrwP7RYXQyRoGC4RAkhAwNe6cGgH4GIpdI9TpIUqiAQts3QXoftPRAJwEykS92wA6GCP\nBJCAxXMkLloFWIbpb4AEIBJAAsLeR5qpgkjQMmEfozDu76ZAJGiZ0M8jGURSxbRL7q4PI/iD\nfQaRNNnSawmjslce/glZx38ViCTCll5LGJW98pDJhtkqiCTCll5LGJW9cqa/y2ZLryWMyl45\nIpXNll5LGJW9ckQqmy29ljAqe+WIVDZbei1hVPbKEalstvRawqjslSNS2WzptYRR2StHpOvG\nQzipkuMaE+H1JwaRkvZCQyDSBERK2gsNgUgTEClpLzQEIk2I3NzRIXNukf7yQLsXmgSRJmxn\nj/QX9kiqINIEREraCw2BSBMQKWkvNAQiTUCkpL3QEIg0AZGS9kJDINIEREraCw2BSBMQKWkv\nNAQiTUCkpL3QEIg0AZGS9kJDINIEREraCw2BSBMQKWkvNAQiTUCkpL3QEIg0AZGS9kJDINIE\nREraCw2BSBMQKWkvNAQiTUCkpL3QEIg0AZGS9kJDINIE/81duHsMIjUGIk3w3lxjeTR5HpHa\nApEmIFJQL8AdRJoQKZLlvoF/yYXXrQsLugNi5SDShHR7pHVtlPTc+kVbBJEmpJtsWNdGSc+t\nX7RFEGlCuunvdW2U9Nz6RVsEkSYgUopFWwSRJiBSikVbBJEmIFKKRVsEkSYg0rWUOfE5RN4z\nCFlpBSOBSEs11i65IUR247E9WdhIINJSjbVLbghEWqawzREHkVaASMsUtjniINIKEGmZwjZH\nHERaASItU9jmiINIK0CkZQrbHHEQaQWItExhmyMOIq0AkZYpbHPEQaQVINIyhW2OOIi0AkRa\nprDNEQeRVoBIyxS2OeIg0goQaZnCNkccRFoBIi1T2OaIg0grQKRlEm2O9bL2cWHIde9r693r\nBD+HSLMg0jJpNsfYVjUutC7gsbKQevf6iJQYRFpmeyIZ9kipQaRllEQKaetFpFVbE/SctbTa\nD2smYtVLFRGphs/NFi9SvnOk1ndTqzpFRiSRRdOiKJJvUxORTEBFn4YQyRtECqF0kUIq+jSE\nSN4gUgh6Inm3hEhlgEghqIkU2MOIlB1ECkFLpNC+QKTsIFIIiRqevbLBXC5BU5OjejmvbEAk\n38KA59cs2ZJI1YBI3iBSCNvMgBtE8gaRQthmBtwgkjeIFMI2M+AmVqSQq1XqQaRT7Cv1X3Sh\n0diVJgaRnKWRha3XT7jvX/U8IolSZGY2WR+RNk2RmdlkfUTaNEVmZpP1EWnTFJmZTdZHpE1T\nZGY2WR+RNk2RmdlkfUTaNEVmZpP1EWnTFJmZTdZHJPW2zUuJfJsepZUHOXd9RFJvG5G2WB+R\ntBs3zwXyTXqVVh7k3PURSbvx7uftIkNzKxsViTTpVVp5kHPXRyTl1od/N4tGDyW2rsjMbLJ+\n9Er9r+lGpLtEtz8QaTv1i9woOUoRqf/PZiySzGdKGsoMImmSV6TLcMf7xwnSZI8k0aB3ae4x\nr7x+kRslRxkijexBpI3UL3Kj5Mgs0niH9BDJjIQSac+rNPeYV16/yI2SowyRrt85cbfIDL8F\nNq6hzCCSJrlF0qahzCCSJojkLM095pXXL3Kj5EAkZ2nuMa+8fpEbJQciOUtzj3nl9YvcKDkQ\nyVmae8wrr1/kRsmBSM7S3GNeef0iN0oORHKW5h7zyusXuVFyIJKzNPeYV16/yI2SA5GcpbnH\nvPL6RW6UHIjkLM095pXXL3Kj5EAkZ2nuMa+8fpEbJQciOUtzj3nl9YvcKDkQyVmae8wrr1/k\nRsmBSM7S3GNeef0iN0oORHKW5h7zyusXuVFyIJKzNPeYV16/yI2SA5GcpbnHvPL6RW6UHIjk\nLM095pXXL3Kj5EAkZ2nuMa+8fpEbJQciOUtzj3nl9YvcKDkQyVmae8wrr1/kRsmBSM7S3GNe\nef0iN0oORHKW5h7zyusXuVFyIJKzNPeYV16/yI2SA5GcpbnHvPL6RW6UHIjkLM095pXXL3Kj\n5EAkZ2nuMa+8fpEbJQciOUtzj3nl9YvcKDkQyVmae8wrr1/kRsnRmkgAIiASQAIQCSABiASQ\ngDZE+m9v/gqv+HdfmWhEDJGm5M5skchFXAeNiCHSlNyZLRK5iOugETFEmpI7s0UiF3EdNCKG\nSFNyZ7ZI5CKug0bEEGlK7swWiVzEddCIGCJNyZ3ZIpGLuA7OXCRMPyJNyZ3ZIpGLuA6uWBhE\nCsS/z3NntkjkIq6DIxWGPVIo/n2eO7NFIhdxHeyhMFs8tDMyDKv37/PcmS0SuYjr4MjcJkWS\nXb1/n+fObJHIRVwHZ+QQKRD/Ps+d2SKRi7gOzsQhUiD+fZ47s0UiF3EdrImbHvxHg0hTcme2\nSOQiroNG7BBpSu7MFolcxHXQiB0iTcmd2SKRi7gOGrFDpCm5M1skchHXQTZcPYg0JXdmi0Qu\n4jrIhqsHkabkzmyRyEVcB9lw9SDSlNyZLRK5iOsgG66eDCLNXswjQ+4kVk5uEWKRDVdPDpE8\nyxKSO4mVk1uEWGTD1YNIsEhuEWKRDVcPIsEiuUWIRTZcPYgEi+QWIRbZcPUgEiySW4RYZMPV\ng0iwSG4RYpENVw8iwSK5RYjFFYs+dWnefUEkWCS3CLE4UmEeH+6Lzx8iwSK5RYjFHgpzQaRQ\nciexcnKLEIsrFlsUSfYuQrmTWDm5RYhlNonbEkmY3EmsnNwixOKKBSKFkjuJlZNbhFhcsUCk\nUHInsXJyixCLKxaIFEruJFZObhFiccUCkULJncTKyS1CLK5YIFIouZNYOblFiMUViw1e2SBM\n7iRWTm4RYtGIGCLBIrlFiEUjYogEi+QWIRaNiEmk2vd6Az3846Ec0YpRliEGjYiJiKTf5AL+\nSVBOY8UoyxCDRsQQaYpyGitGWYYYNCKGSFOU01gxyjLEoBExRJqinMaKUZYhBo2IIdIU5TRW\njLIMMWhEDJGmKKexYpRliEEjYog0RTmNFaMsQwz2UKR9KwaRpiinsWKUZYjBHoq0OUSkKcpp\nrBhlGWKwhwKRwvFPgnIaK0ZZhhismUgcw9JFErz5ibXPldNYMcoyxOBOVkgUF4KablW+6wwT\nKWZD7vgnQTmNFaMsQwzuZLUz2YBIpaIsQwzS8Uq7Ju91ItImUJYhBul4pV2T9zoRaRMoyxCD\nO1mIFIZ/EpTTWDHKMsTgTBaTDYH4J0E5jRWjLEMMjmgxaxeMfxKU01gxyjLEkCRDCyDSFOU0\nVoyyDDEkydACiDRFOY0VoyxDDEkytAAiTVFOY8UoyxBDkgwtkEOkVVf5RJE7dFsktx0BJMnQ\nAhlEyrCu3KHbIrntCCBJhhZAJFhHbjsCSJKhBRAJ1pHbjgCSZGgBRIJ15LYjgCQZWgCRYB25\n7QggSYYWQCRYR247ArAnq/qbn2RYV+7QbZHcdgQgHa+0a5JZJyKVSm47AhBOV+pViawTkUol\ntx0BCKdruqq4u4wIXI2QYrOGteQO3RbJbUcAi+FPEVSBlWa439YCuUO3RXLbEYArFogUSu7Q\nbZHcdgTgigUihZI7dFsktx0BOFKRNKeIBOvIbUcAjlQgUjC5Q7dFctsRgCMViBRM7tBtkdx2\nBOBIBSIFkzt0WyS3HQFoRAyRYB257QhAI2KIBOvIbUcAGhFDJFhHbjsC0IhYGyJZuzd3EjeJ\nsiJ+aEQMkSAlyor4oRExRIKUKCvih0bEEAlSoqyIHxoRQyRIibIifmhEDJEgJcqK+KERMUSC\nlCgr4ocrFubpdwyIBClRVsQPRyqGT1An+Ug3IkFKlBXxwx4KcwuqYY/kjbV7c2dukygr4ocr\nFhs8tEtz0xXnzU+s3Zs7c5tEWRE/loK6JZHStW7D2r25M7dJlBXxYyl1iOSLtXtzZ26TKCvi\nx1LqEMkXa/fmztwmUVbEj6XUIZIv1u7NnblNoqyIH0upQyRfrN2bO3ObRFkRP5ZSh0i+WLs3\nd+Y2ibIifiylDpF8sXZv7sxtEmVF/JANVw8iQUqUFfFDNlw9iAQpUVbED9lw9SASpERZET9k\nw9UjIlLExTwy5I5X4yCSGoi0ZRBJDUTaMoikBiJtGURSA5G2DCKpgUhbBpHUQKQtg0hqINKW\nKVWk2z0bkrz9gkggTqEimeHmJylAJBCnTJEedxFKASKBOGWKdEtdouiVIpIMw+pzJ6lxyhYp\nzRVqhYgkTO4kNU7ZIt1/RIFIIE7RIj09WgsigTiItBVyJ6lxihaJQ7sAciepcYoXickGT3In\nqXGKFunCrJ0/uZPUOKWKlBJEAnEQaSvkTlLjINJWyJ2kehAJsl5LdjQihkgwRiTIei3Z0YgY\nIsEYkSDrtWRHI2KIBGNEgqzXkh2NiCESjBEJsl5LdjQihkgwRiTIei3Z0YgYIsEYkSDrtWRH\nI2KIBGNEgqzXkh1XLLhEKBTlNFaMSJD1WrLjSIXh6u9QlNNYMSJB1mvJjj0Uho9RBKOcxooR\nCbJeS3ZcsdjgB/u4+UkhiARZryU7ztRZHq1OcLpVxcDtuApBJMh6LdlZSN2WJhsQqRBEgqzX\nkp3l1LFH8kM5jRUjEmS9luwspw6R/FBOY8WIBFmvJTuzqdvWrB0iFYJIkPVasjObOkQKQDmN\nFSMSZL2W7MynblNXNiBSIYgEWa8lO7Lh6kEkGCMSZL2W7MiGqweRYIxIkPVasiMbrh5EgjEi\nQdZryY5suHokRIq5mEcG/+FtHZEg525fNlw9IiKpVAkhdnjbQSTIuduXDVcPIsEYkSDnbl82\nXD2IBGNEgpy7fdlw9SASjBEJcu72ZcPVg0gwRiTIuduXDVcPIsEYkSDnbt+RuaSTxYgEY0SC\nnLt9d+TS5Q6RYIxIkHO3744cIoURO7ztIBLk3O1rRK4UkWQYVh87vO0gEuTc7TtSusVzJGFi\nh7cdRIKcu31rJq4Wbe7QTpjY4W0HkSDnbt+aiW2eIwkTO7ztIBLk3O1bM4FIK4gd3nYQCXLu\n9q2ZQKQVxA5vO4gEOXf71kwg0gpih7cdRIKcu317KDZ5ZYMwscPbDiJBzt2+RsQQCcaIBDl3\n+xoRQyQYIxLk3O1rRAyRYIxIkHO3rxGxGZHir8wphtiBDM5jtYgEOfeWakRsTqTodRZD7PCI\nJKFIkoij0n0B7WtEDJF8hkckCUWSRByV7gtoXyNiiOQzPCJJKJIk4qh0X0D7GhFDJJ/hEUlC\nkSQRR6X7AtrXiBgi+QyPSBKKJIk4Kt0X0L5GxBDJZ3hEklAkScRR6b6A9jUihkg+wyOShCJJ\nIo5K9wW0bw+F2rV2iCSbhCJJIo5K9wW0b82E3tXfiCSbhCJJIo5K9wW0b80EIq0gdnhEklAk\nScRR6b6A9q2ZqF+kJPcHCrtWKXZ4RJJQJEnEUem+gPbdKa1bpLXrXU/s8IgkoUiSiKPSfQHt\nWzNxvYkQIoUROzwiSSiSJOKodF9A+/ZQdEcsiBRG7PCIJKFIkoij0n0B7buDgUhhxA6PSBKK\nJIk4Kt0X0L41E5wjrSB2eESSUCRJxFHpvoD2rZlApBXEDo9IEookiTgq3RfQvj0U1V/ZgEgF\nk0Qcle4LaF8jYojkMzwiSSiSJOKodF9A+xoRQySf4RFJQpEkEUel+wLa14iYiEie1xvoITKS\nm6QekQLQiJiESL5N6pF7JOsBkVaCSDAGkVaCSDAGkVaCSDAGkVaCSDAGkVaCSDAGkVaCSDCm\nKZH6y+3SvCWDSDCmJZFG+iCSH7lHsh4aEsnMhT8YRIIxDYk0G/5gcoikx9Bk7pGshyZFSpH0\nDCJlIPdI1gMirQSRYAwirQSRYEyLIiUJOiLBGERaCSLBGERaCSLBGERaCSLBmKZESgkiwRhE\n8uJzf7n87M3+z70EkWAMIvlw6t7r33Vv+d9NQiQYg0g+HMzX5dvsL1/mMBQhUrMUmXkRkuep\n2yF9m/fxBeSI1CyItJrOn6M5IRL8FZEiOJjvk9ldOLSDvyJSBKdunuGj2yGdhiJEahZEWs/n\nrjtDuuy/7iWI1CyIlBJEapbWRUp7G3pEapbGRTL3H2s4Ha8zdz/T1dlXiUjbBpHWZ/zQ787M\n7m4SIjVL4yJdWZnxT3M4dyJ9mreXNeUViZuf6INIqzO+M+f+vdji3pAVbiv3SBYJIq2O3fWw\nDpHgCiKtjt3+tkfqLlx9XhUitQYirU7d7RzptDOfL+tCpNZoXqSI0B1vJ+H3S+0QqV1aFykq\nc937SOb4uEIIkdqlcZGm87rRIFKzNC5SYhCpWRApgq/u0O7t9ChApGZBpPUcbpMNx3sJIjUL\nIq3m3ey6nRHT341hVWabzlhJnqed+b7+5g3ZtkCkxNxn+7hEqCkQKTHv9z3ScShCpAZApNR8\nXM+R/uy4sqEpECkxlk/rIFIDIFJiEKlNEEkeRGoAREqcuv3Hz3MRIjUAItlDtzp1v4dzzy4h\nUgMgkj1zq1N3/np7dgmRGgCRBFL352M/dqkUkWQYVp97JDODSI7URab2u/vSvs+ndbFH2i6I\nJJK602H0cXNEagBESp+688fv7mh/Ov/adJysC5G2CyKlTt2fbrLhvb/gjjdkmwGREqeum2b4\nPA9r2U3WhUjbBZESp84cTy9F7pUi0lZApISp647kzpZi90oRaSsgUkIc10MgUgMgUkIQqV0Q\nKSGI1ASNK2MnaZYQqQkQyULSLDmuQkOkbYFIFpJmCZGaAJEsJM0Sh3ZNgEgWkmYJkZoAkSwk\nzRIiNQEiWUiaJURqAkSy4MicWXXTBkRqAkSy4I5cutwh0rZAJAvuyCFSGLlHUg9EsqCRu1JE\n4uYnaUAkCzO5S5dg9yrZI1UIIllwhS7hl5oj0sZAJAsauUOkbYFIFjRyh0jbApEsuCO3Indc\ntNoEiGTBHTlECiP3SOqBSBYcmWOyIZjcI6kHIlmQDVcPIm0LRLKQNEsc2jUBIllImiVEagJE\nsiAbrh5Eqhac8UU2XD2IVC2I5ItAok7H7qju+PjuS0SqFkTyJX2gDv3pkdndTUKkakEkX5Ln\n6dMczp1In+ZtKEKkakEkX5LnaWfO/UfOmbXbAIjkS/I8XQ/rEGkjIJIvjsytv0Rof9sjfZv9\nfW0vDy7uEjkQKRxE8sUduZW5u50jnXbmc7I6+yoRqWwQyRd35Nbm7ni7ruEwXZ19lYhUNojk\niztyq3PXvY9kjl9Pq7OvkpuflA0i+WKP3P1HmgS/PHh9SgH2SOEgki+OzPF5pGByj6QIiOSL\nO3Krcvfzdp1iOO8/R4WIVC2I5Is7cmty97Mzx+73yYyuEEKkekEkX9yRW5O7vXk7Xx/8OTze\nRkKkekEkX9yRW5G7k/m4Pz6a+7wdIlULIvniyNy6yYY3c74//nm8kYRI1YJIviTN0sQ+rrXb\nAIjkS9Is7RBpYyCSL0mz9GZO98enfv6uA5GqBZF8SZql78ek98+OyYaqsCqDM96kDdO72X18\n//7+/thx0WpdIFIcidP0cb+W8+1RiEgVgEhxpI7Tz3t385Pjx8+oDJEqAJHikA1XDyJVACLF\nIRuuHkSqAESKQzZcPYhUAYgUh2y4ehCpAhApjvnUpfmAHyJVACLF4Qrd4wLw+PwhUgUgUhyu\nzCFSKLlHMhJEimM5ddsRibsIzYBIcThTZ3m0OsHuVbFHKgVEimMxdUw2eJJ7JCNBpDgWU4dI\nnuQeyUgQKY6l1KVIHyJVACLFsZC6JOFDpApApDjmU5cme4hUAYgUx2zqpvO7q0GkCkCkOGTD\n1YNIFYBIcciGqweRKgCR4pANVw8ilQXKCCAbrh5EKgtEEkA2XD2IVBaIJIBsuHoQqSwQSQDZ\ncPUgUlkgkgCy4ep5iOT+EILmZsiQeyT9QSQBZMPVM5NgRMoAIgkgG64eRCoLRBJgMXUpP0YR\n9FRyEOkGIgngCp15flLKxZYAAA2WSURBVBABIpUFIgngypx5ehADIpUFIgkwnzqzqUM7bn7S\ng0gCOFM3/NyQSLKrzz2S/iCSALOpM2nih0hlgUgCzKUu0f0hEakwEEmAudSluvgAkcoCkQRY\nTB17JE9yj6Q/iCTAYuoQyZPcI+kPIgmwmDpE8iT3SPqDSALIhqsHkcoCkQSQDVcPIpUFIgkg\nG64eRMoGzmghG64eRMoGImkhG64eRMoGImkhG64eRMoGImkhG64eRMoGImkhG64eRMoGImkh\nG64eRMoGImkhG64eRMoGImkxn7o0d55DpGwgkhau0D0+i8THKPzIPZJWEEkLV+YQKZTcI2kF\nkbSYTd3GRGrw5ieIpIUzddMfcQle9VRy2CMhkhyzqes/bx4dMUTKBiJpMZ+67sgFkfzIPZJW\nEEmL5dQhkh+5R9IKImkxmzrOkQLIPZJWEEmL2dQhUgC5R9IKImkxnzqubPAn90haQSQtZMPV\ng0jZQCQtZMPVg0jZQCQtZMPVg0jZQCQtZMPVg0gKWJXBGTVkw9WDSAogUl5kw9WDSAogUl5k\nw9WDSAogUl5kw9WDSAogUl5kw9WDSAogUl5kw9WDSAogUl4cmbtdGsQlQv5kHkhEyos7coaL\nVoPIPJCIlBd35BApjMwDiUh5mc3dxkTa9M1PECkvM7nbnEiyq888kIiUl9nYIVIAmQcSkfIy\nGztECiDzQCJSXmZTh0gBZB5IRMrLbOgQKYDMA4lIeZnNHCIFkHkgESkv9sgN87pc2eBP5oFE\npLzIhqsHkRRApLzIhqsHkRRApLzIhqsHkRRApLzIhqsHkdKCMgUiG64eREoLIhWIbLh6ECkt\niFQgsuHqQaS0IFKByIarB5HSgkgFIhuuHkRKCyIViGy4ehApLYhUILLh6kGktCBSgTgyx12E\ngtEbM0QqEHfk+Kh5GHpjhkgF4o4cIoWhN2aIVCDLuduOSFu5ixAiFchM7p5+RyR41VPJYY8E\ncizGjskGT/TGDJEKZDF2iOSJ3pghUoEspS5F+hApLYhUIAuhSxI+REoLIhXIfObSZA+R0oJI\nBWKP3G1edzq/uxpESgsiFYhsuHoQKS2IVCCy4epBpLQgUoHIhqsHkVaDM7UgG64eRFoNItWC\nbLh6EGk1iFQLsuHqQaTVIFItyIarB5FWg0i1IBuuHkRaDSLVgmy4ehBpNYhUC7Lh6kGk1SBS\nLTgyx81PghEZHkSqBXfkuGdDGCLDg0i14I4cIoUhMjyIVAvLuatSJKEbnajf/ASRamE543WK\npNjWDZHhQaRacOaij12tkw2IBLo4c2FeHqwGkVaDSLXgioWxPFoLIq0GkWrBkQrz+IlIfogM\nDyLVgj0UZvQLkfwQGR5EqgV7DId53WqvbEAk0EU2XD2ItBpEqgXZcPUg0moQqRZkw9WDSKtB\npFqQDVcPIvmAM1UjG64eRPIBkapGNlw9iOQDIlWNbLh6EMkHRKoa2XD1IJIPiFQ1suHqQSQf\nEKlqZMPVg0g+IFLV2EMxujSIS4Q8iR0JRKoadwyHT/bFRwyRfECkqnHH8PYDkTyJHQlEqhp3\nMMzlcSehOCRE8r0lSUCVzDc/QaSqmc94wSJFPS1B7EggUtU4c2FG/yJBJB8QqWqcuTDjU6U4\nEMkHRKoaVyyGz8em+IgsIvmASFXjSEXKmwghkheIVDX2UBjrw7Ugkg+IVDXWTIwP6RDJk9iR\nQKSq0YgYIvmASFWjETFE8gGRqkYjYojkAyJVjUbEEGmKVZnWnam9TzQihkhTEMlC7X2iETFE\nmoJIFmrvE42IIdIURLJQe59oRAyRpiCShdr7RCNiiDQFkSzU3icaEUOkKYhkofY+0YgYIk1B\nJAu194krFsOtT0q9+QkibYza+8SRCjO+/0ksiDQFkSzU3if2UJgLIoXi3+eIZKH2PnHFItXH\nzBdWE9LC4m2CnMuKMjTp3+eIZKH2PplNuLmUdI5knH8UgX+fI5KF2vvEFYtepDS3EUKkKYhk\nofY+ccWiwHMkRNoytfeJKxaIFIp/nyOShdr7xBULRArFv88RyULtfeKKBSKF4t/niGSh9j5x\nxaLAKxsQacvU3icaEUOkKYhkofY+0YgYIk1BJAu194lGxBoWqfZ4tEPASFkX1YgYIiFS8SBS\nKcQOD2QFkUohdnggK4hUCrHDA1lBpFKIHR7ICiKVQuzwQFYQqRRihweyIiUSlwiFEjs8kBUh\nkUq8+QkigRwyIhV58xNEAjmE9khqIq25zUhwRVGGLYodHshK5SJlWI0UscMDWREVSX6yIcNq\npIgdHsgKe6RSiB0eyAoilULs8EBWEKkUYocHsoJIpRA7PJAVUZGYbAggdnggK1IipQSREKl4\nEKkUYocHsoJIpRA7PJAVRCoFnEmPtU8DOjW2fsBaNSKGSLAORJqASLAORJqASLAORJqASLAO\nRJqASLAORJqASLCO6kV6XBqUIr2IBOuoXaTRFauI5AsipWczIhn2SN4gUnpqF+mKGf2MI5VI\nZTJsHiKlB5Esa9o6iJSeLYhU3mRD4SBSehDJtqqNg0jp2YBI5ul3DIgE66hfJPPyIAJEgnVU\nL5KxPFoPIsE6ahdpNK+LSL4gUnpqFykxiATrQKQJiATrQKQJiATrQKQJbYgEIAwiASQAkQAS\ngEgACUAkgAQgEkACEAkgAYgEkABEAkgAIgEkAJEAEoBIAAlAJIAEIBJAAhAJIAGIBJAARAJI\nACIBJACRABKASAAJQCSABDQt0vVWm+PvUeoLb3fhtCzvUf1ivX/nyzc2OQtnmrKt1NG+Y61e\nK3Utatl+S0/Z+3TS3Pw2WUut9Wf6Tz/WLYtkpj8ehS/xsnx3maP66FtJnxd9Wam9vrsp20od\n7T+X2rCutPsiyNeKAT3l3KhHc/Mb5fLipb69U23dp0EbItm71z8e1m92mwny9ZeXSKP2lpqy\nrtRPJLedzyvtt+c5hdZNXSOS10a9SuBu6rnQ2n0atCGS/WUGiGT7P35WpItZHHOHSM6mrCv1\n3CPZXr91pXe9LM4lFmluoyz/5XmKZOs+DRoRyT1oNpEs6e5PCByrtKZj+Sjk/tO/Kc9Dy5et\nsr1+60pn/suxifSy+Y5OMWa0Dq+Nmpba6jteqaX7NGhFJCuOyQbHIfbL18gPp+qv43hb/nWl\nry291J9r6nWltlNw21bZsK/UsahlU6095Wre9VKfF3NvrLP9lzov3adA0yIFonj+qn2qvDEy\ndB8DBpCARkS6HwCsfL3W+vaVBiwax8I5mudkh9dGxdb3J+BFBdRXoA2R+vNiy3l9TH37SgMW\njcV2iuMo9S8MaCqgvj8BLyqgvjwNiXT9ZZ0BmMyFeb/l4lip96KON7e8N2pQ8/W03lLqXRjS\nlKVQ8UUF1RenKZGep3OsL95WaK1vX+mKRZfbdw6TfX7KWupbGNTUa6HiiwqrL0xbIr389zu7\n8EJ9+0oDFk0TurhVzL3YdfUVX1RYfVnaEMn+lklsfftKAxaNR2CPFNSUyH/+AS8qoL4wjYi0\nTSTOkUKaEjkdCXhRAfXFQaR6se/l7LtJ78KApkT2sgEvKqC+PG2IZCyPHKWRha3XL3KjNGhD\nJJn/vGX2CHr/+duJPZu0Vhd5Ubl2PlaK2AgFRE4nZM5R9E5H7EQ2pfiiMp0OWSljKzQQmeCS\nmTXTmyCzE9mU4ovKMkFnpZTtUMD+Uq2lkYWK9WWIbErxRRWT32I2RBz2SN6wRwqnlO2QhnMk\nbzhHWkMZWyFO7FxSwKyTXn0ZIptSfFGKnbJMERshjrE8cpRGFirWlyGyKcUXpdgpHpSwDQDV\ng0gACUAkgAQgEkACEAkgAYgEkABEAkgAIgEkAJEAEoBIAAlAJIAEIBJAAhAJIAGIBJAARAJI\nACIBJACRABKASAAJQCSABCASQAIQCSABiASQAEQCSAAiASQAkQASgEgACUAkgAQgEkACEAkg\nAYgEkABEAkgAItXK99vOvJ2eS7vvr7t/h50Zobx1zUEHV8p778f+Z1qMSJmgg+vkw+x+90bn\n319TkyYi3UtAHHq5Sn4Ggd7M2+QJRMoEvVwl7+ajf3A+fna/Pvdmd33gEuls9pfh92/Ju9m9\n90/fa15OB2MOLydd4AciVcnBfI//PF5Pgw6XmT3S0fzpfn39GmjMx7D4qOZnfy71qbL92wOR\nqmTqyskczpfzwZxmRDr1x4Bv5ue3ZPd9+d6Zr0nNXefm123HBaEgUpVMXTma86U7ajvOnSPt\nrwt1opjOm1+Hjk81OayLAJGq5NmV+xy3W6TP7rTqT/fjVtIveq/5bszxe3LACAEgUpUc7+dI\np7OnSGez62bNf1wiXT52v7+fZtPBF0Sqko9h1u5Pf6x2f2Jm+vv999htv788iTRe9PS+5xxp\nJYhUJff3kQ7dNNvxcXozI9K3OXxf/TPXCbzr7MPx+cSId51WQr/Vydv1yoafY3e8dvnqpuF+\nT4KO82/I7s2uO7K7z9qdJjX33Swes3ZrQaRKOYyvtev/2P3Mi3QyvSbGXJc/Tmt+9Sv8o/ga\ntgQi1crX8VeIr9sfn3tj3vq9jVuks+nPrH5Ljmb/+VSzv7IBj1aCSO3wu0e6GsN5kAD0aTsc\nbtf/IJIA9GkrDFfXIZII9Gkr7PrZhQsiiUCfAiQAkQASgEgACUAkgAT8f4Yne6bMt2oLAAAA\nAElFTkSuQmCC",
      "text/plain": [
       "Plot with title \"Color Key\n",
       "and Histogram\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(\"DoubletDecon\")\n",
    "location=LS_nal_no_mor/\" #Update as needed \n",
    "expressionFile=paste0(location, \"counts.txt\")\n",
    "genesFile=paste0(location, \"Top50Genes.txt\")\n",
    "clustersFile=paste0(location, \"Cluster.txt\")\n",
    "\n",
    "newFiles=Seurat_Pre_Process(expressionFile, genesFile, clustersFile)\n",
    "\n",
    "results=Main_Doublet_Decon(rawDataFile=newFiles$newExpressionFile, \n",
    "                           groupsFile=newFiles$newGroupsFile, \n",
    "                           filename=\"cntl\", \n",
    "                           location=location,\n",
    "                           fullDataFile=NULL, \n",
    "                           removeCC=FALSE, \n",
    "                           species=\"mmu\", \n",
    "                           rhop=1.1, \n",
    "                           write=TRUE, \n",
    "                           PMF=TRUE, \n",
    "                           useFull=FALSE, \n",
    "                           heatmap=FALSE,\n",
    "                           centroids=TRUE,\n",
    "                           num_doubs=100, \n",
    "                           only50=TRUE,\n",
    "                           min_uniq=4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# checking doublet results and barcode "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "LS_sal.data <-Read10X(data.dir = \"/LS-SAL-1/raw_feature_bc_matrix\")\n",
    "colnames(LS_sal.data) = paste0(colnames(LS_sal.data),\"LS_sal\")\n",
    "LS_sal<- CreateSeuratObject(counts = LS_sal.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_sal@meta.data$stim <- \"LS_sal\"\n",
    "\n",
    "LS_sal_clean<-rownames(read.table(file=\"/Users/stuberadmin/Documents/scRNAseq/201130/Doublet_processing/LS_sal/Final_nondoublets_groups_cntl.txt\"))\n",
    "LS_sal_doublet<-rownames(read.table(file=\"/Users/stuberadmin/Documents/scRNAseq/201130/Doublet_processing/LS_sal/Final_doublets_groups_cntl.txt\"))\n",
    "doubletrate_LS_sal<-100*length(LS_sal_doublet)/(length(LS_sal_clean)+length(LS_sal_doublet))\n",
    "doubletrate_LS_sal\n",
    "LS_sal<-subset(x=LS_sal,cells=LS_sal_clean)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_mor.data <-Read10X(data.dir = \"/LS-MOR-1/raw_feature_bc_matrix\")\n",
    "colnames(LS_mor.data) = paste0(colnames(LS_mor.data),\"LS_mor\")\n",
    "LS_mor<- CreateSeuratObject(counts = LS_mor.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_mor@meta.data$stim <- \"LS_mor\"\n",
    "\n",
    "LS_mor_clean<-rownames(read.table(file=\"/Doublet_processing/LS_mor/Final_nondoublets_groups_cntl.txt\"))\n",
    "LS_mor_doublet<-rownames(read.table(file=\"/Doublet_processing/LS_mor/Final_doublets_groups_cntl.txt\"))\n",
    "doubletrate_LS_mor<-100*length(LS_mor_doublet)/(length(LS_mor_clean)+length(LS_mor_doublet))\n",
    "doubletrate_LS_mor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "LS_nal.data <-Read10X(data.dir = \"/LS-NAL-1/raw_feature_bc_matrix\")\n",
    "colnames(LS_nal.data) = paste0(colnames(LS_nal.data),\"LS_nal\")\n",
    "LS_nal<- CreateSeuratObject(counts = LS_nal.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_nal@meta.data$stim <- \"LS_nal\"\n",
    "\n",
    "LS_nal_clean<-rownames(read.table(file=\"/Doublet_processing/LS_nal/Final_nondoublets_groups_cntl.txt\"))\n",
    "LS_nal_doublet<-rownames(read.table(file=\"/Doublet_processing/LS_nal/Final_doublets_groups_cntl.txt\"))\n",
    "doubletrate_LS_nal<-100*length(LS_nal_doublet)/(length(LS_nal_clean)+length(LS_nal_doublet))\n",
    "doubletrate_LS_nal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "10.6580166821131"
      ],
      "text/latex": [
       "10.6580166821131"
      ],
      "text/markdown": [
       "10.6580166821131"
      ],
      "text/plain": [
       "[1] 10.65802"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "LS_one_mor.data <-Read10X(data.dir = \"/Mor_single/analysis\")\n",
    "colnames(LS_one_mor.data) = paste0(colnames(LS_one_mor.data),\"LS_one_mor\")\n",
    "LS_one_mor<- CreateSeuratObject(counts = LS_one_mor.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_one_mor@meta.data$stim <- \"LS_one_mor\"\n",
    "\n",
    "LS_one_mor_clean<-rownames(read.table(file=LS_one_mor/Final_nondoublets_groups_cntl.txt\"))\n",
    "LS_one_mor_doublet<-rownames(read.table(file=LS_one_mor/Final_doublets_groups_cntl.txt\"))\n",
    "doubletrate_LS_one_mor<-100*length(LS_one_mor_doublet)/(length(LS_one_mor_clean)+length(LS_one_mor_doublet))\n",
    "doubletrate_LS_one_mor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "7.48506132718314"
      ],
      "text/latex": [
       "7.48506132718314"
      ],
      "text/markdown": [
       "7.48506132718314"
      ],
      "text/plain": [
       "[1] 7.485061"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "LS_nal_no_mor.data <-Read10X(data.dir = \"/Nal_no_mor/analysis\")\n",
    "colnames(LS_nal_no_mor.data) = paste0(colnames(LS_nal_no_mor.data),\"LS_nal_no_mor\")\n",
    "LS_nal_no_mor<- CreateSeuratObject(counts = LS_nal_no_mor.data, min.cells = 3, min.features = 200, project = \"10X_LS_morphine\")\n",
    "LS_nal_no_mor@meta.data$stim <- \"LS_nal_no_mor\"\n",
    "\n",
    "LS_nal_no_mor_clean<-rownames(read.table(file=LS_nal_no_mor/Final_nondoublets_groups_cntl.txt\"))\n",
    "LS_nal_no_mor_doublet<-rownames(read.table(file=LS_nal_no_mor/Final_doublets_groups_cntl.txt\"))\n",
    "doubletrate_LS_nal_no_mor<-100*length(LS_nal_no_mor_doublet)/(length(LS_nal_no_mor_clean)+length(LS_nal_no_mor_doublet))\n",
    "doubletrate_LS_nal_no_mor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
